<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="jhmarcus" />

<meta name="date" content="2019-03-04" />

<title>Human Origins Array Global Data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">drift-workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Human Origins Array Global Data</h1>
<h4 class="author"><em>jhmarcus</em></h4>
<h4 class="date"><em>2019-03-04</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-03-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>drift-workflow/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190211code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190211)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190211code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190211)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjhmarcusdriftworkflowtree17bb44232c274e9c03d926e5eed30021026033datargetblank17bb442a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jhmarcus/drift-workflow/tree/17bb44232c274e9c03d926e5eed30021026033da" target="_blank">17bb442</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjhmarcusdriftworkflowtree17bb44232c274e9c03d926e5eed30021026033datargetblank17bb442a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/flash_cache/
    Ignored:    data.tar.gz
    Ignored:    data/datasets/
    Ignored:    data/raw/
    Ignored:    output.tar.gz
    Ignored:    output/admixture/
    Ignored:    output/admixture_benchmark/
    Ignored:    output/benchmark/
    Ignored:    output/flash_greedy/
    Ignored:    output/log/
    Ignored:    output/sim/

Unstaged changes:
    Modified:   analysis/data_hoa_global.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/17bb44232c274e9c03d926e5eed30021026033da/analysis/data_hoa_global.Rmd" target="_blank">17bb442</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-05
</td>
<td>
addded miss per pop and contributer
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/17bb44232c274e9c03d926e5eed30021026033da/docs/data_hoa_global.html" target="_blank">17bb442</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-05
</td>
<td>
addded miss per pop and contributer
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/e2a3abaac502e13c82c48278b8e11359f710591c/analysis/data_hoa_global.Rmd" target="_blank">e2a3aba</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-05
</td>
<td>
updated plink filtering command
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/e2a3abaac502e13c82c48278b8e11359f710591c/docs/data_hoa_global.html" target="_blank">e2a3aba</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-05
</td>
<td>
updated plink filtering command
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/9a69c08cb1787c854dd8a16b8a13c173b8b3466b/analysis/data_hoa_global.Rmd" target="_blank">9a69c08</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
<td>
updated data hoa
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/9a69c08cb1787c854dd8a16b8a13c173b8b3466b/docs/data_hoa_global.html" target="_blank">9a69c08</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
<td>
updated data hoa
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/f8154d84d278ccefb1ce4e07724cf65ad3adfe73/analysis/data_hoa_global.Rmd" target="_blank">f8154d8</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
<td>
added to data rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/a1580ed016dba24c34c736e2cd013724d76c72e6/analysis/data_hoa_global.Rmd" target="_blank">a1580ed</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
<td>
added data exploration
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/a1580ed016dba24c34c736e2cd013724d76c72e6/docs/data_hoa_global.html" target="_blank">a1580ed</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
<td>
added data exploration
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here I explore basic properties of the Human Origins Array dataset. I downloaded the data from:</p>
<p><a href="https://reich.hms.harvard.edu/sites/reich.hms.harvard.edu/files/inline-files/NearEastPublic.tar.gz" class="uri">https://reich.hms.harvard.edu/sites/reich.hms.harvard.edu/files/inline-files/NearEastPublic.tar.gz</a></p>
<p>I subsequently converted the <code>eigenstrat</code> files to <code>plink</code> format using the following parameter file and <code>convertf</code> command:</p>
<pre><code>genotypename:   HumanOriginsPublic2068.geno
snpname:    HumanOriginsPublic2068.snp
indivname:  HumanOriginsPublic2068.ind
outputformat:   PACKEDPED
genotypeoutname:    HumanOriginsPublic2068.bed
snpoutname: HumanOriginsPublic2068.bim
indivoutname:   HumanOriginsPublic2068.fam
familynames:    NO</code></pre>
<pre><code>convertf -p eig2plink.par</code></pre>
<p>I then removed the sex chromosomes using the the following <code>plink</code> command:</p>
<pre><code>plink --bfile HumanOriginsPublic2068 --make-bed --autosome --out HumanOriginsPublic2068_auto</code></pre>
<div id="imports" class="section level1">
<h1>Imports</h1>
<p>Lets import some needed packages:</p>
<pre class="r"><code>library(ggplot2)
library(tidyr)
library(dplyr)
library(lfa)</code></pre>
</div>
<div id="read-genotypes" class="section level1">
<h1>Read Genotypes</h1>
<p>Here I read the full genotype matrix of the Human Origins dataset:</p>
<pre class="r"><code>Y = t(lfa:::read.bed(&quot;../data/raw/NearEastPublic/HumanOriginsPublic2068_auto&quot;))</code></pre>
<pre><code>[1] &quot;reading in 2068 individuals&quot;
[1] &quot;reading in 616938 snps&quot;
[1] &quot;snp major mode&quot;
[1] &quot;reading snp 20000&quot;
[1] &quot;reading snp 40000&quot;
[1] &quot;reading snp 60000&quot;
[1] &quot;reading snp 80000&quot;
[1] &quot;reading snp 100000&quot;
[1] &quot;reading snp 120000&quot;
[1] &quot;reading snp 140000&quot;
[1] &quot;reading snp 160000&quot;
[1] &quot;reading snp 180000&quot;
[1] &quot;reading snp 200000&quot;
[1] &quot;reading snp 220000&quot;
[1] &quot;reading snp 240000&quot;
[1] &quot;reading snp 260000&quot;
[1] &quot;reading snp 280000&quot;
[1] &quot;reading snp 300000&quot;
[1] &quot;reading snp 320000&quot;
[1] &quot;reading snp 340000&quot;
[1] &quot;reading snp 360000&quot;
[1] &quot;reading snp 380000&quot;
[1] &quot;reading snp 400000&quot;
[1] &quot;reading snp 420000&quot;
[1] &quot;reading snp 440000&quot;
[1] &quot;reading snp 460000&quot;
[1] &quot;reading snp 480000&quot;
[1] &quot;reading snp 500000&quot;
[1] &quot;reading snp 520000&quot;
[1] &quot;reading snp 540000&quot;
[1] &quot;reading snp 560000&quot;
[1] &quot;reading snp 580000&quot;
[1] &quot;reading snp 600000&quot;</code></pre>
<pre class="r"><code># number of individuals
n = nrow(Y)

# number of SNPs
p = ncol(Y)</code></pre>
</div>
<div id="missingness-per-snp" class="section level1">
<h1>Missingness per SNP</h1>
<p>Here I compute the missingness per SNP:</p>
<pre class="r"><code>n_miss_snp = colSums(is.na(Y))
p_snpmss = qplot(n_miss_snp / n, bins=100) + theme_bw() + 
           scale_x_continuous(breaks = pretty(n_miss_snp / n, n = 10)) +
           xlab(&quot;Missingness Fraction&quot;) +
           ylab(&quot;Count&quot;)

p_snpmss</code></pre>
<p><img src="figure/data_hoa_global.Rmd/miss-per-snp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-miss-per-snp-1">
Past versions of miss-per-snp-1.png
</button>
</p>
<div id="fig-miss-per-snp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/17bb44232c274e9c03d926e5eed30021026033da/docs/figure/data_hoa_global.Rmd/miss-per-snp-1.png" target="_blank">17bb442</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/9a69c08cb1787c854dd8a16b8a13c173b8b3466b/docs/figure/data_hoa_global.Rmd/miss-per-snp-1.png" target="_blank">9a69c08</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/a1580ed016dba24c34c736e2cd013724d76c72e6/docs/figure/data_hoa_global.Rmd/miss-per-snp-1.png" target="_blank">a1580ed</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>There are very few SNPs with high levels of missing data so we can use a very stringent missingness threshold without losing much information.</p>
</div>
<div id="missingness-per-individual" class="section level1">
<h1>Missingness per Individual</h1>
<p>Here I compute the missingness per individual:</p>
<pre class="r"><code>n_miss_ind = rowSums(is.na(Y))
p_indmss = qplot(n_miss_ind / p) + theme_bw() +
           xlab(&quot;Missingness Fraction&quot;) +
           ylab(&quot;Count&quot;)
p_indmss</code></pre>
<p><img src="figure/data_hoa_global.Rmd/miss-per-ind-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-miss-per-ind-1">
Past versions of miss-per-ind-1.png
</button>
</p>
<div id="fig-miss-per-ind-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/17bb44232c274e9c03d926e5eed30021026033da/docs/figure/data_hoa_global.Rmd/miss-per-ind-1.png" target="_blank">17bb442</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/9a69c08cb1787c854dd8a16b8a13c173b8b3466b/docs/figure/data_hoa_global.Rmd/miss-per-ind-1.png" target="_blank">9a69c08</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/a1580ed016dba24c34c736e2cd013724d76c72e6/docs/figure/data_hoa_global.Rmd/miss-per-ind-1.png" target="_blank">a1580ed</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>It seems like a few individuals are missing about 20000 of their SNPs which is a bit worrisome maybe they should be removed from the analysis? For now I will in include them and see if they pop up as any outliers in the PCs.</p>
</div>
<div id="missingness-per-population" class="section level1">
<h1>Missingness per Population</h1>
<p>Here I compute the missingness fraction per population:</p>
<pre class="r"><code># meta data
meta_df = read.table(&quot;../data/meta/HumanOriginsPublic2068.meta&quot;, sep=&quot;\t&quot;, header=T)
meta_df$miss_frac = n_miss_ind / p

# average missingness per pop for sorting
avg_miss_df = meta_df %&gt;% 
              group_by(Simple.Population.ID) %&gt;% 
              summarise(avg_miss=mean(miss_frac)) %&gt;%
              arrange(desc(avg_miss)) 

# distribution of missingness per pop
p_popmss = ggplot(meta_df, aes(x=factor(Simple.Population.ID, 
                                        levels=avg_miss_df$Simple.Population.ID), 
                               y=miss_frac)) + 
           geom_boxplot() +
           theme_classic() +
           theme(axis.text.x = element_text(angle = 90, hjust = 1, size=6)) +
           xlab(&quot;Population&quot;) +
           ylab(&quot;Missingness Fraction&quot;)
p_popmss</code></pre>
<p><img src="figure/data_hoa_global.Rmd/miss-per-pop-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-miss-per-pop-1">
Past versions of miss-per-pop-1.png
</button>
</p>
<div id="fig-miss-per-pop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/17bb44232c274e9c03d926e5eed30021026033da/docs/figure/data_hoa_global.Rmd/miss-per-pop-1.png" target="_blank">17bb442</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="missingness-per-contributer" class="section level1">
<h1>Missingness per Contributer</h1>
<p>Here I compute the average missingness fraction per contributor:</p>
<pre class="r"><code># average missingness per contributer for sorting
avg_miss_df = meta_df %&gt;% 
              group_by(Contributor) %&gt;% 
              summarise(avg_miss=mean(miss_frac)) %&gt;%
              arrange(desc(avg_miss)) 

# distribution of missingness per contributer
p_conmss = ggplot(meta_df, aes(x=factor(Contributor, 
                                        levels=avg_miss_df$Contributor), 
                               y=miss_frac)) + 
           geom_boxplot() +
           theme_classic() +
           theme(axis.text.x = element_text(angle = 90, hjust = 1, size=6)) +
           xlab(&quot;Contributer&quot;) +
           ylab(&quot;Missingness Fraction&quot;)
p_conmss</code></pre>
<p><img src="figure/data_hoa_global.Rmd/miss-per-con-1.png" width="816" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-miss-per-con-1">
Past versions of miss-per-con-1.png
</button>
</p>
<div id="fig-miss-per-con-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/17bb44232c274e9c03d926e5eed30021026033da/docs/figure/data_hoa_global.Rmd/miss-per-con-1.png" target="_blank">17bb442</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>It seems like there is some variation in the amount of missingness per pop and contributor (there might be some confounding there) but the total amount of missingness is so low I think it can be ignored?</p>
</div>
<div id="filter" class="section level1">
<h1>Filter</h1>
<p>Given the above results here are the <code>plink</code> commands I ran to filter the data:</p>
<pre><code>plink --bfile HumanOriginsPublic2068 --geno .005 --maf .05 --make-bed --autosome --out HumanOriginsPublic2068_auto_maf05_geno005</code></pre>
These filtering steps take us from 616938 to 343758 SNPs … which still likely contains a lot of information about population structure. <br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS  10.14.2

Matrix products: default
BLAS/LAPACK: /Users/jhmarcus/miniconda3/lib/R/lib/libRblas.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] lfa_1.12.0    dplyr_0.8.0.1 tidyr_0.8.2   ggplot2_3.1.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0       compiler_3.5.1   pillar_1.3.1     git2r_0.23.0    
 [5] plyr_1.8.4       workflowr_1.2.0  tools_3.5.1      digest_0.6.18   
 [9] evaluate_0.12    tibble_2.0.1     gtable_0.2.0     pkgconfig_2.0.2 
[13] rlang_0.3.1      yaml_2.2.0       xfun_0.4         withr_2.1.2     
[17] stringr_1.4.0    knitr_1.21       fs_1.2.6         rprojroot_1.3-2 
[21] grid_3.5.1       tidyselect_0.2.5 glue_1.3.0       R6_2.4.0        
[25] rmarkdown_1.11   purrr_0.3.0      corpcor_1.6.9    magrittr_1.5    
[29] whisker_0.3-2    backports_1.1.3  scales_1.0.0     htmltools_0.3.6 
[33] assertthat_0.2.0 colorspace_1.4-0 labeling_0.3     stringi_1.2.4   
[37] lazyeval_0.2.1   munsell_0.5.0    crayon_1.3.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
