<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="jhmarcus" />

<meta name="date" content="2019-02-11" />

<title>Non-Negative FLASH for visualizing Population Structure</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">drift-workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Non-Negative FLASH for visualizing Population Structure</h1>
<h4 class="author"><em>jhmarcus</em></h4>
<h4 class="date"><em>2019-02-11</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-03-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>drift-workflow/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190211code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190211)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190211code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190211)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
flash-setup
</li>
<li>
soft-setup
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>flash_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjhmarcusdriftworkflowtree7392ef2a88404e40b5c1b1d6981c8428a5094025targetblank7392ef2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jhmarcus/drift-workflow/tree/7392ef2a88404e40b5c1b1d6981c8428a5094025" target="_blank">7392ef2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjhmarcusdriftworkflowtree7392ef2a88404e40b5c1b1d6981c8428a5094025targetblank7392ef2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/flash_cache/
    Ignored:    data.tar.gz
    Ignored:    data/datasets/
    Ignored:    data/raw/
    Ignored:    output.tar.gz
    Ignored:    output/admixture/
    Ignored:    output/admixture_benchmark/
    Ignored:    output/flash_greedy/
    Ignored:    output/log/
    Ignored:    output/sim/

Untracked files:
    Untracked:  sandbox/admixture-benchmark.Rmd
    Untracked:  sandbox/sgdp.Rmd

Unstaged changes:
    Deleted:    analysis/admixture-benchmark.Rmd
    Modified:   code/viz.R
    Modified:   sandbox/hoa_weurasia.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/analysis/flash.Rmd" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
<td>
updated to autosomes and added manhatten plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/flash.html" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
<td>
updated to autosomes and added manhatten plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/38a461d4a7d655238d6508eaf62f6daeee605252/docs/flash.html" target="_blank">38a461d</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
<td>
built all
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/403bc6becb8b3a3bf9c901aa0cb3b9df258ac14e/docs/flash.html" target="_blank">403bc6b</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added hide code blocks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/b4749ac6e4cb22e7da9e07c2f03a1baacbf48cb2/docs/flash.html" target="_blank">b4749ac</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
fixed some typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/7a2b6c7d6a6827a391b617d74eb09098b1873a9a/docs/flash.html" target="_blank">7a2b6c7</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added backfit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/f5ef1af4d17d9469ed52811c88166186d0c3c993/analysis/flash.Rmd" target="_blank">f5ef1af</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added workflows for human origins datasets
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/f5ef1af4d17d9469ed52811c88166186d0c3c993/docs/flash.html" target="_blank">f5ef1af</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added workflows for human origins datasets
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/31a16c694bf65cba8fb598420548cc747aa27abc/analysis/flash.Rmd" target="_blank">31a16c6</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-14
</td>
<td>
added hists of factors
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/31a16c694bf65cba8fb598420548cc747aa27abc/docs/flash.html" target="_blank">31a16c6</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-14
</td>
<td>
added hists of factors
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/95385101c23660b8db007cd135b90085db6b22c3/analysis/flash.Rmd" target="_blank">9538510</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-13
</td>
<td>
switched to struct style plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/95385101c23660b8db007cd135b90085db6b22c3/docs/flash.html" target="_blank">9538510</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-13
</td>
<td>
switched to struct style plot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/63c2b7528f4e2498d7fd7c795958509858a268ee/analysis/flash.Rmd" target="_blank">63c2b75</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-13
</td>
<td>
added softimpute
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/63c2b7528f4e2498d7fd7c795958509858a268ee/docs/flash.html" target="_blank">63c2b75</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-13
</td>
<td>
added softimpute
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/7b6ecbcc9fbf6e03e40cd5ceb3c224b5ac18b44e/analysis/flash.Rmd" target="_blank">7b6ecbc</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-12
</td>
<td>
made fig size bigger
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/7b6ecbcc9fbf6e03e40cd5ceb3c224b5ac18b44e/docs/flash.html" target="_blank">7b6ecbc</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-12
</td>
<td>
made fig size bigger
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/5c0f91aa44dadc88f2410d991fd5b8c215652340/analysis/flash.Rmd" target="_blank">5c0f91a</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-12
</td>
<td>
init flash proof of principle
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/5c0f91aa44dadc88f2410d991fd5b8c215652340/docs/flash.html" target="_blank">5c0f91a</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-12
</td>
<td>
init flash proof of principle
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here I explore Matthew’s suggestion of applying a version of <code>FLASH</code> to visualize population structure. Let <span class="math inline">\(\mathbf{Y}\)</span> be the <span class="math inline">\(n \times p\)</span> genotype matrix where the rows are individuals and the columns are SNPs. The elements of this matrix <span class="math inline">\(y_{ij} \in \{0, 1, 2\}\)</span> encode the count of an arbitrarily predefined allele for the diploid individual <span class="math inline">\(i\)</span> and SNP <span class="math inline">\(j\)</span>. Here we imagine that the genotypes can be explained by <span class="math inline">\(K\)</span> latent drift events. Each drift event effects the entire genome, i.e. all of the SNPs, but only a subset the individuals’ ancestors were in a “population” whom experienced the <span class="math inline">\(k\)</span>th event. For instance, the ancestors of all Europeans experienced the “Out of Africa” drift event and the ancestors of North and South Americans crossed the Bering strait, undergoing a bottleneck. Here is a figure from a recent review (Nielson et al. 2017) that describes some of the major “drift events” in human history:</p>
<p><img src="../data/img/nielsen_et_al_2017.png" /></p>
<p>We assume the following matrix factorization model for the genotypes that can flexibly describe this kind of process:</p>
<p><span class="math display">\[
y_{ij} = \mu_j + \sum_{k=1}^K \ell_{ik} \delta_{jk} + e_{ij}
\]</span></p>
<p>where <span class="math inline">\(\mu_j\)</span> represents the mean genotype at the <span class="math inline">\(j\)</span>th SNP (something like a mean allele frequency), <span class="math inline">\(\ell_{ik}\)</span> represents a non-negative weight of the <span class="math inline">\(i\)</span>th individual on the <span class="math inline">\(K\)</span>th drift event, <span class="math inline">\(\delta_{jk}\)</span> represents a deviance from the mean genotype which defines the drift event, and <span class="math inline">\(e_{ij} \sim N(0, \sigma^2)\)</span> is a random error with variance <span class="math inline">\(\sigma^2\)</span>. Given the model I described above it would be natural for <span class="math inline">\(\boldsymbol{\ell}_k\)</span> to be sparse, with different levels of sparsity for each <span class="math inline">\(k\)</span> i.e. in a dataset of global Human diversity, all the non-African individuals experienced the “Out of Africa” drift event but only a subset of samples experienced the founding of Finland, or some population specific processes. Here we assume the following priors for the individual weights (loadings) and deviances (factors):</p>
<p><span class="math display">\[
\ell_{1k}, \dots, \ell_{nk} \overset{iid}{\sim} g_k, \quad  g_k \in \mathcal{G}_+ \\
\delta_{jk} \overset{iid}{\sim} N(0, 1)
\]</span></p>
<p>Briefly: where <span class="math inline">\(g_k\)</span> is an adaptive shrinkage prior that is constrained to be in the family of unimodal and non-negative distributions <span class="math inline">\(\mathcal{G}_+\)</span>. The prior on the sample weights can be fit using <code>flashier</code> using an efficient approach that adapts to unknown sparsity using Empirical Bayes via solving iterative convex optimization problems. We assume the deviances come from a dense prior given that all SNPs are effected by the drift event. More can be expanded upon the model later, lets now explore a proof of principle with a real dataset using <code>flashier</code>.</p>
<div id="imports" class="section level2">
<h2>Imports</h2>
<p>Lets import some needed packages</p>
<pre class="r"><code>library(ggplot2)
library(tidyr)
library(dplyr)
library(softImpute)
library(flashier)</code></pre>
</div>
<div id="functions" class="section level2">
<h2>Functions</h2>
<pre class="r"><code>positive_structure_plot = function(gath_df, pops, K_, label_size=5){
  
  p = ggplot(data=gath_df, aes(x=reorder(iid, desc(value)), y=value, fill=factor(K, levels=2:K_))) + 
      geom_bar(stat=&quot;identity&quot;, width=1) +  
      scale_fill_brewer(palette = &quot;Set3&quot;) + 
      scale_y_continuous(expand=c(0, 0)) +
      scale_x_discrete(expand=c(0, 0)) +
      facet_grid(. ~ factor(clst, levels=pops), scales = &quot;free&quot;, space=&quot;free&quot;, switch=&quot;both&quot;) + 
      theme_classic() +
      theme(panel.spacing = unit(0.2, &quot;lines&quot;), 
            strip.background = element_rect(colour=&quot;white&quot;, fill=&quot;white&quot;),
            strip.text.x = element_text(size = label_size, colour = &quot;black&quot;, angle = 90, hjust = 1.1), 
            strip.placement = &quot;outside&quot;, 
            panel.grid.major = element_blank(), 
            panel.grid.minor = element_blank(),
            axis.text.x=element_blank(), 
            axis.ticks.x=element_blank()) + 
            ylab(&quot;&quot;) + 
            xlab(&quot;&quot;) + 
            guides(fill=F)
  
  return(p)
  
}</code></pre>
</div>
<div id="data" class="section level2">
<h2>Data</h2>
<p>I prepared a dataset described in <a href="https://reich.hms.harvard.edu/sites/reich.hms.harvard.edu/files/inline-files/nature19310_article_0.pdf">Lazaridis et al. 2016</a> which includes 2068 individuals and 621799 SNPs sampled from around the globe! This dataset is often referred to as the “Human Origins Array Dataset” (HoA) and is commonly used a reference panel to determine the ancestry of individuals from a new sample of human genetic variation. I removed all SNPs with allele frequency less than 5% and with sample missingness fraction greater than 1%. I then LD pruned the resulting genotype matrix using standard parameters in <code>plink</code>. This resulted in 167178 SNPs which will help us do some exploratory analysis more efficiently. First we read this filtered dataset:</p>
<pre class="r"><code>X = t(lfa:::read.bed(&quot;../data/raw/NearEastPublic/HumanOriginsPublic2068_maf_geno_auto_ldprune&quot;))</code></pre>
<pre><code>[1] &quot;reading in 2068 individuals&quot;
[1] &quot;reading in 165468 snps&quot;
[1] &quot;snp major mode&quot;
[1] &quot;reading snp 20000&quot;
[1] &quot;reading snp 40000&quot;
[1] &quot;reading snp 60000&quot;
[1] &quot;reading snp 80000&quot;
[1] &quot;reading snp 100000&quot;
[1] &quot;reading snp 120000&quot;
[1] &quot;reading snp 140000&quot;
[1] &quot;reading snp 160000&quot;</code></pre>
<pre class="r"><code>n = nrow(X)
p = ncol(X)
print(n)</code></pre>
<pre><code>[1] 2068</code></pre>
<pre class="r"><code>print(p)</code></pre>
<pre><code>[1] 165468</code></pre>
<p>Next we read in some meta data that includes population and regional labels for each individual:</p>
<pre class="r"><code># read the meta data
meta_df = read.table(&quot;../data/raw/NearEastPublic/HumanOriginsPublic2068_clst.tsv&quot;, sep=&quot;\t&quot;)
colnames(meta_df) = c(&quot;iid&quot;, &quot;clst&quot;)
head(meta_df)</code></pre>
<pre><code>     iid    clst
1 SA1004 Khomani
2  SA063 Khomani
3  SA010 Khomani
4  SA064 Khomani
5  SA073 Khomani
6 SA1025 Khomani</code></pre>
<pre class="r"><code># read the clst data
clst_df = read.table(&quot;../data/raw/NearEastPublic/meta.tsv&quot;, sep=&quot;\t&quot;, header=TRUE)
clst_df$clst = clst_df$Simple.Population.ID
clst_df = clst_df %&gt;% distinct(clst, .keep_all = TRUE) 
head(clst_df)</code></pre>
<pre><code>  Simple.Population.ID Verbose.Population.ID Region  Country Latitude
1        Ain_Touta_WGA         Ain_Touta_WGA Africa  Algeria     35.4
2             Algerian              Algerian Africa  Algeria     36.8
3             Mozabite              Mozabite Africa  Algeria       32
4              BantuSA       Bantu_SA_Ovambo Africa   Angola      -19
5                 Gana                  Gana Africa Botswana    -21.7
6                  Gui                   Gui Africa Botswana    -21.5
  Longitude Samples Passed.QC                   Contributor          clst
1       5.9       3         0 Mark G. Thomas / Leila Laredj Ain_Touta_WGA
2         3       7         7                   David Comas      Algerian
3         3      27        21         Patterson et al. 2012      Mozabite
4        18       1         1         Patterson et al. 2012       BantuSA
5      23.4       9         8 Pickrell et al. 2012 and 2014          Gana
6      23.3      11         7 Pickrell et al. 2012 and 2014           Gui</code></pre>
<p>To start lets sub-sample the genotype matrix so we have fewer individuals</p>
<pre class="r"><code>set.seed(12345)
n_samples = 500
idx = sample(n, n_samples, replace = FALSE)
Y =  X[idx, ]
m = nrow(Y)
p = ncol(Y)
print(m)</code></pre>
<pre><code>[1] 500</code></pre>
<pre class="r"><code>print(p)</code></pre>
<pre><code>[1] 165468</code></pre>
</div>
<div id="flash-flashier" class="section level2">
<h2>FLASH (<code>flashier</code>)</h2>
<p>Next lets run greedy <code>flashier</code> on the resulting data matrix with a maximum of 10 factors:</p>
<pre class="r"><code>K = 9 # the first factor is fixed

# to start we use point.normal not normal
flash_res = flashier(Y, 
                     greedy.Kmax=K, 
                     prior.type=c(&quot;nonnegative&quot;, &quot;point.normal&quot;), 
                     var.type=0,
                     fix.dim=list(1), 
                     fix.idx=list(1:m), 
                     fix.vals=list(rep(1, m)))</code></pre>
<pre><code>Initializing flash object...
Adding factor 1 to flash object...
Adding factor 2 to flash object...
Adding factor 3 to flash object...
Adding factor 4 to flash object...
Adding factor 5 to flash object...
Adding factor 6 to flash object...
Adding factor 7 to flash object...
Adding factor 8 to flash object...
Adding factor 9 to flash object...
Adding factor 10 to flash object...
Nullchecking 9 factors...
Wrapping up...
Done.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>It seems like the objective decreases in many of the factor fits (is this something to worry about). Lets now visualize the fitted <code>flashier</code> fit:</p>
<pre class="r"><code># setup loadings data.frame
l_df = as.data.frame(flash_res$loadings$normalized.loadings[[1]])
l_df$iid = as.vector(meta_df$iid[idx]) # individual ids
l_df$clst = meta_df$clst[idx] # population labels
l_df$lab = substr(l_df$clst, 1, 3) # shorthand for population labels

# join with the meta data
l_df = l_df %&gt;% inner_join(clst_df, on=&quot;clst&quot;)
l_df = l_df %&gt;% arrange(Region, clst) # sort by region then by population
l_df$iid = factor(l_df$iid, levels = l_df$iid) # make sure the ids are sorted
colnames(l_df)[1:(K+1)] = 1:(K+1)

# gather the data.frame for plotting
l_gath_df = l_df %&gt;% 
            select(-Simple.Population.ID, -Verbose.Population.ID, -Country, 
                   -Latitude, -Longitude, -Samples, -Passed.QC, -Contributor) %&gt;%
            gather(K, value, -iid, -clst, -lab, -Region) %&gt;% 
            filter(K!=1)

#### viz #####
pops = unique(l_df$clst)

# Africa
africa_df = clst_df %&gt;% filter(Region == &quot;Africa&quot;, Simple.Population.ID %in% pops) %&gt;% arrange(desc(Latitude))
africa_pops = africa_df$Simple.Population.ID
p_africa = positive_structure_plot(l_gath_df %&gt;% filter(Region == &quot;Africa&quot;), africa_pops, 10, label_size=5)

# America
america_df = clst_df %&gt;% filter(Region == &quot;America&quot;, Simple.Population.ID %in% pops) %&gt;% arrange(desc(Latitude))
america_pops = america_df$Simple.Population.ID
p_america = positive_structure_plot(l_gath_df %&gt;% filter(Region == &quot;America&quot;), america_pops, 10, label_size=5) 

# Central Asia Siberia
central_asia_siberia_df = clst_df %&gt;% filter(Region == &quot;CentralAsiaSiberia&quot;, Simple.Population.ID %in% pops) %&gt;% arrange(desc(Latitude))
central_asia_siberia_pops = central_asia_siberia_df$Simple.Population.ID
p_central_asia_siberia = positive_structure_plot(l_gath_df %&gt;% filter(Region == &quot;CentralAsiaSiberia&quot;), central_asia_siberia_pops, 10, label_size=5)

# East Asia
east_asia_df = clst_df %&gt;% filter(Region == &quot;EastAsia&quot;, Simple.Population.ID %in% pops) %&gt;% arrange(desc(Latitude))
east_asia_pops = east_asia_df$Simple.Population.ID
p_east_asia = positive_structure_plot(l_gath_df %&gt;% filter(Region == &quot;EastAsia&quot;), east_asia_pops, 10, label_size=5)

# South Asia
south_asia_df = clst_df %&gt;% filter(Region == &quot;SouthAsia&quot;, Simple.Population.ID %in% pops) %&gt;% arrange(desc(Latitude))
south_asia_pops = south_asia_df$Simple.Population.ID
p_south_asia= positive_structure_plot(l_gath_df %&gt;% filter(Region == &quot;SouthAsia&quot;), south_asia_pops, 10, label_size=5)

# West Eurasia
west_eurasia_df = clst_df %&gt;% filter(Region == &quot;WestEurasia&quot;, Simple.Population.ID %in% pops) %&gt;% arrange(desc(Latitude))
west_eurasia_pops = west_eurasia_df$Simple.Population.ID
p_west_eurasia = positive_structure_plot(l_gath_df %&gt;% filter(Region == &quot;WestEurasia&quot;), west_eurasia_pops, 10, label_size=5)

# Oceania
oceania_df = clst_df %&gt;% filter(Region == &quot;Oceania&quot;, Simple.Population.ID %in% pops) %&gt;% arrange(desc(Latitude))
oceania_pops = oceania_df$Simple.Population.ID
p_oceania = positive_structure_plot(l_gath_df %&gt;% filter(Region == &quot;Oceania&quot;), oceania_pops, 10, label_size=5) 

p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = &quot;v&quot;) 
p</code></pre>
<p><img src="figure/flash.Rmd/flash-viz-loadings-1.png" width="816" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-viz-loadings-1">
Past versions of flash-viz-loadings-1.png
</button>
</p>
<div id="fig-flash-viz-loadings-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/flash.Rmd/flash-viz-loadings-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/31a16c694bf65cba8fb598420548cc747aa27abc/docs/figure/flash.Rmd/flash-viz-loadings-1.png" target="_blank">31a16c6</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here each row of the facet grid represents a different latent “drift event”. Each tick on the x-axis is a different individual. I represent each individual by a three letter short hand for the population label provided in the meta data. I then color each individual by a broad scale regional level label. There is much that can be improved in the visualization (which I like to think about more as its quite important) but I think this is a good starting place that helps reveal some patterns. Here are a couple observations:</p>
<ul>
<li>As one would hope (as I set it as an argument in <code>flashier</code>) the 1st drift event is a constant value across the individuals</li>
<li>The 2nd drift event looks like it represents “Out of Africa”. Interestingly it picks up a number of American samples as being weighted on this event. If we look more closely they have the label “AA” which is shorthand for African American and as such these individuals carry African ancestry</li>
<li>The 4th and 6th drift event looks like its related to the peopling of the Americas</li>
<li>The 5th drift event seems to be related to the migration into Ocenia i.e. Australia / Papua New Guinea. This might be wishful thinking but its interesting to see the small weight of SouthAsian populations on this event.</li>
<li><code>TODO: describe more events</code></li>
</ul>
<p>My impression is that this approach could be promising in building an interpretable visualization of population structure. Lets now visualize the factors … the question here is to see if FLASH estimated them to be sparse or not b/c I used a point normal prior on the deviances (not a normal distribution as described above):</p>
<pre class="r"><code># read factors
delta_df = as.data.frame(flash_res$loadings$normalized.loadings[[2]])
colnames(delta_df)[1:(K+1)] = 1:(K+1) 

# gather the data.frame for plotting
delta_gath_df = delta_df %&gt;% 
                gather(K, value) %&gt;%
                filter(K!=1)

# plot the factors
K_ = K + 1
p_fct = ggplot(delta_gath_df, aes(x=value, fill=factor(K, 2:K_))) + 
        scale_fill_brewer(palette = &quot;Set3&quot;) +
        geom_histogram() + 
        facet_wrap(~factor(K, levels=2:K_), scales = &quot;free&quot;) + 
        labs(fill=&quot;K&quot;) + 
        theme_bw()
p_fct</code></pre>
<p><img src="figure/flash.Rmd/flash-viz-factors-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-viz-factors-1">
Past versions of flash-viz-factors-1.png
</button>
</p>
<div id="fig-flash-viz-factors-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/flash.Rmd/flash-viz-factors-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/31a16c694bf65cba8fb598420548cc747aa27abc/docs/figure/flash.Rmd/flash-viz-factors-1.png" target="_blank">31a16c6</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Interestingly, some of the latent drift events seems be denser than others … so it seems the <code>point.normal</code> prior is doing some magic here. It will take some reading of the tea leaves to understand why some of the drift events are sparser than others. It seems like the later factors tend to be sparser. What I like about this plot is that it kind of serves a natural legend for the colors in the loadings plot.</p>
</div>
<div id="pca-softimpute" class="section level2">
<h2>PCA (<code>softImpute</code>)</h2>
<p>Here we center and scale the columns and use <code>softImpute</code> to compute a PCA like solution in spite of missing data:</p>
<pre class="r"><code>Z = biScale(Y, row.center=F, row.scale=F)
soft_res = softImpute(Z, rank.max=9, type=&quot;svd&quot;)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Here we visualize the result:</p>
<pre class="r"><code># setup loadings data.frame
l_df = as.data.frame(soft_res$u)
l_df$iid = as.vector(meta_df$iid[idx]) # individual ids
l_df$clst = meta_df$clst[idx] # population labels
l_df$lab = substr(l_df$clst, 1, 3) # shorthand for population labels

# join with the meta data
l_df = l_df %&gt;% inner_join(clst_df, on=&quot;clst&quot;)
l_df = l_df %&gt;% arrange(Region, clst) # sort by region then by population
l_df$iid = factor(l_df$iid, levels = l_df$iid) # make sure the ids are sorted
colnames(l_df)[1:K] = paste0(&quot;K=&quot;, 1:K) # name the columns

# gather the data.frame for plotting
l_gath_df = l_df %&gt;% 
            select(-Simple.Population.ID, -Verbose.Population.ID, -Country, 
                   -Latitude, -Longitude, -Samples, -Passed.QC, -Contributor) %&gt;%
            gather(K, value, -iid, -clst, -lab, -Region) 

# setup loadings data.frame
l_df = as.data.frame(soft_res$u)
l_df$iid = as.vector(meta_df$iid[idx]) # individual ids
l_df$clst = meta_df$clst[idx] # population labels
l_df$lab = substr(l_df$clst, 1, 3) # shorthand for population labels

# join with the meta data
l_df = l_df %&gt;% inner_join(clst_df, on=&quot;clst&quot;)
l_df = l_df %&gt;% arrange(Region, clst) # sort by region then by population
l_df$iid = factor(l_df$iid, levels = l_df$iid) # make sure the ids are sorted
colnames(l_df)[1:K] = paste0(&quot;K=&quot;, 1:K) # name the columns

# gather the data.frame for plotting
l_gath_df = l_df %&gt;% 
            select(-Simple.Population.ID, -Verbose.Population.ID, -Country, 
                   -Latitude, -Longitude, -Samples, -Passed.QC, -Contributor) %&gt;%
            gather(K, value, -iid, -clst, -lab, -Region) 

# plot facet grid
p = ggplot(data=l_gath_df, aes(x=iid, y=value, label=lab, color=Region)) + 
    geom_text(size=2.5) +  
    scale_colour_brewer(palette = &quot;Set2&quot;, guide=guide_legend(override.aes=list(size=4))) + 
    theme_bw() +
    theme(axis.text.x=element_blank(), axis.ticks.x=element_blank()) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    facet_grid(factor(K, levels=paste0(&quot;K=&quot;, 1:9))~., scales=&quot;free_y&quot;) +
    xlab(&quot;Individual&quot;) + 
    ylab(&quot;Loading&quot;) 
p</code></pre>
<p><img src="figure/flash.Rmd/soft-viz-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-soft-viz-1">
Past versions of soft-viz-1.png
</button>
</p>
<div id="fig-soft-viz-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/flash.Rmd/soft-viz-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/63c2b7528f4e2498d7fd7c795958509858a268ee/docs/figure/flash.Rmd/soft-viz-1.png" target="_blank">63c2b75</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<code>softImpute</code>’s solution looks pretty good and picks up similar patterns it seems, though its a bit harder to interpret negative / positive loadings and some of the drift events look denser which also makes interpretation trickier. I’m also now convinced by the power of the stacked bar plot (STRUCTURE plot)! <br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS  10.14.2

Matrix products: default
BLAS/LAPACK: /Users/jhmarcus/miniconda3/lib/R/lib/libRblas.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] flashier_0.1.0 softImpute_1.4 Matrix_1.2-15  dplyr_0.8.0.1 
[5] tidyr_0.8.2    ggplot2_3.1.0 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0         lfa_1.12.0         RColorBrewer_1.1-2
 [4] compiler_3.5.1     pillar_1.3.1       git2r_0.23.0      
 [7] plyr_1.8.4         workflowr_1.2.0    tools_3.5.1       
[10] digest_0.6.18      evaluate_0.12      tibble_2.0.1      
[13] gtable_0.2.0       lattice_0.20-38    pkgconfig_2.0.2   
[16] rlang_0.3.1        yaml_2.2.0         xfun_0.4          
[19] withr_2.1.2        stringr_1.4.0      knitr_1.21        
[22] fs_1.2.6           cowplot_0.9.4      rprojroot_1.3-2   
[25] grid_3.5.1         tidyselect_0.2.5   glue_1.3.0        
[28] R6_2.4.0           rmarkdown_1.11     reshape2_1.4.3    
[31] corpcor_1.6.9      purrr_0.3.0        magrittr_1.5      
[34] whisker_0.3-2      backports_1.1.3    scales_1.0.0      
[37] htmltools_0.3.6    assertthat_0.2.0   colorspace_1.4-0  
[40] labeling_0.3       stringi_1.2.4      lazyeval_0.2.1    
[43] munsell_0.5.0      crayon_1.3.4      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
