<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="jhmarcus" />

<meta name="date" content="2019-02-15" />

<title>Human Origins Array Global Results</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">drift-workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Human Origins Array Global Results</h1>
<h4 class="author"><em>jhmarcus</em></h4>
<h4 class="date"><em>2019-02-15</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-03-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>drift-workflow/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190211code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190211)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190211code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190211)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjhmarcusdriftworkflowtree17bb44232c274e9c03d926e5eed30021026033datargetblank17bb442a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jhmarcus/drift-workflow/tree/17bb44232c274e9c03d926e5eed30021026033da" target="_blank">17bb442</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjhmarcusdriftworkflowtree17bb44232c274e9c03d926e5eed30021026033datargetblank17bb442a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/flash_cache/
    Ignored:    data.tar.gz
    Ignored:    data/datasets/
    Ignored:    data/raw/
    Ignored:    output.tar.gz
    Ignored:    output/admixture/
    Ignored:    output/admixture_benchmark/
    Ignored:    output/benchmark/
    Ignored:    output/flash_greedy/
    Ignored:    output/log/
    Ignored:    output/sim/

Unstaged changes:
    Modified:   analysis/data_hoa_global.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/9a69c08cb1787c854dd8a16b8a13c173b8b3466b/analysis/hoa_global.Rmd" target="_blank">9a69c08</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
<td>
updated data hoa
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/9a69c08cb1787c854dd8a16b8a13c173b8b3466b/docs/hoa_global.html" target="_blank">9a69c08</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-04
</td>
<td>
updated data hoa
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/hoa_global.html" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/7392ef2a88404e40b5c1b1d6981c8428a5094025/analysis/hoa_global.Rmd" target="_blank">7392ef2</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
<td>
wflow_publish(&quot;analysis/*.Rmd&quot;)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/analysis/hoa_global.Rmd" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
<td>
updated to autosomes and added manhatten plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/hoa_global.html" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
<td>
updated to autosomes and added manhatten plot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/d01c17cc7176a06b72f1ca5fa1b649f099676f4c/analysis/hoa_global.Rmd" target="_blank">d01c17c</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
<td>
added chrom exploration
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/5ee97edb372e04a524d19b5afce1d54aed3371ea/analysis/hoa_global.Rmd" target="_blank">5ee97ed</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-27
</td>
<td>
updating manhatten plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/e1b4f85025e99ff931cd2b1c2fee626b2c21d176/analysis/hoa_global.Rmd" target="_blank">e1b4f85</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-25
</td>
<td>
added snakemake rule
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/e1b4f85025e99ff931cd2b1c2fee626b2c21d176/docs/hoa_global.html" target="_blank">e1b4f85</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-25
</td>
<td>
added snakemake rule
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/38a461d4a7d655238d6508eaf62f6daeee605252/docs/hoa_global.html" target="_blank">38a461d</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
<td>
built all
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/b5aafbb646dc86e64316f81d9a9e4dd972eed734/docs/hoa_global.html" target="_blank">b5aafbb</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/63e7173f9a33d2a087e6f16d90cd929eb422bb6b/analysis/hoa_global.Rmd" target="_blank">63e7173</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
<td>
expanded upon mean variance
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/63e7173f9a33d2a087e6f16d90cd929eb422bb6b/docs/hoa_global.html" target="_blank">63e7173</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
<td>
expanded upon mean variance
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/analysis/hoa_global.Rmd" target="_blank">38b57c5</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
<td>
simplified greedy flash global analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/38b57c54997e44c495298de7423104f1749299a1/docs/hoa_global.html" target="_blank">38b57c5</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
<td>
simplified greedy flash global analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/403bc6becb8b3a3bf9c901aa0cb3b9df258ac14e/analysis/hoa_global.Rmd" target="_blank">403bc6b</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added hide code blocks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/403bc6becb8b3a3bf9c901aa0cb3b9df258ac14e/docs/hoa_global.html" target="_blank">403bc6b</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added hide code blocks
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/b4749ac6e4cb22e7da9e07c2f03a1baacbf48cb2/analysis/hoa_global.Rmd" target="_blank">b4749ac</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
fixed some typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/b4749ac6e4cb22e7da9e07c2f03a1baacbf48cb2/docs/hoa_global.html" target="_blank">b4749ac</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
fixed some typos
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/7a2b6c7d6a6827a391b617d74eb09098b1873a9a/analysis/hoa_global.Rmd" target="_blank">7a2b6c7</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added backfit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/7a2b6c7d6a6827a391b617d74eb09098b1873a9a/docs/hoa_global.html" target="_blank">7a2b6c7</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added backfit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/f5ef1af4d17d9469ed52811c88166186d0c3c993/analysis/hoa_global.Rmd" target="_blank">f5ef1af</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added workflows for human origins datasets
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jhmarcus/drift-workflow/f5ef1af4d17d9469ed52811c88166186d0c3c993/docs/hoa_global.html" target="_blank">f5ef1af</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
added workflows for human origins datasets
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/4afc77e7c21b668a5b240c920c563dab67acc0e5/analysis/hoa_global.Rmd" target="_blank">4afc77e</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
<td>
init hoa global analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This is an analysis of the full Human Origins dataset which includes 2068 sampled from around the world. I filtered out rare variants with global minor allele frequency less than 5%, removed any variants with a missingness level greater than 1%, and removed any SNPs on the sex chromosomes. I then LD pruned the SNPs using standard parameters in <code>plink</code>, resulting in 165468 SNPs.</p>
<div id="imports" class="section level1">
<h1>Imports</h1>
<p>Lets import some needed packages:</p>
<pre class="r"><code>library(ggplot2)
library(tidyr)
library(dplyr)
library(RColorBrewer)
library(biomaRt)
library(knitr)
source(&quot;../code/viz.R&quot;)</code></pre>
</div>
<div id="flash-greedy" class="section level1">
<h1>FLASH-Greedy</h1>
<p>Here is the <code>snakemake</code> rule I used for running <code>flashier</code>:</p>
<pre><code>    run:
        R(&quot;&quot;&quot;
          # read the genotype matrix
          Y = t(lfa:::read.bed(&#39;{params.bed}&#39;))
          # number of individuals
          n = nrow(Y)
          
          # run greedy flash
          flash_fit = flashier::flashier(Y,
                                         greedy.Kmax=20,
                                         ash.param=list(method=&#39;fdr&#39;),
                                         prior.type=c(&#39;nonnegative&#39;, &#39;normal.mixture&#39;),
                                         var.type=2,
                                         fixed.factors=flashier::ones.factor(1),
                                         output.lvl=4,
                                         verbose.lvl=2)
          # save the rds
          saveRDS(flash_fit, &#39;{output.rds}&#39;)</code></pre>
<p>Lets first read the greedy <code>flashier</code> fit</p>
<pre class="r"><code># read the flash fit output by snakemake
flash_fit = readRDS(&quot;../output/flash_greedy/hoa_global_ld/HumanOriginsPublic2068_maf_geno_auto_ldprune.rds&quot;)

# read the snp meta data
bim_df = read.table(&quot;../data/datasets/hoa_global_ld/HumanOriginsPublic2068_maf_geno_auto_ldprune.bim&quot;, header=F)
colnames(bim_df) = c(&quot;chrom&quot;, &quot;rsid&quot;, &quot;cm&quot;, &quot;pos&quot;, &quot;a1&quot;, &quot;a2&quot;)

# drift event loadings lfsr
l_lfsr_df = data.frame(flash_fit$lfsr[[1]])
colnames(l_lfsr_df) = 1:21

# drift event lfsr
delta_lfsr_df = data.frame(flash_fit$lfsr[[2]])
colnames(delta_lfsr_df) = 1:21
delta_lfsr_df$chrom = bim_df$chrom
delta_lfsr_df$pos = bim_df$pos
delta_lfsr_df$rsid = bim_df$rsid

# drift events
delta_df = as.data.frame(flash_fit$loadings$normalized.loadings[[2]])
colnames(delta_df) = 1:21
delta_df$chrom = bim_df$chrom
delta_df$pos = bim_df$pos
delta_df$rsid = bim_df$rsid

# number of drift events
K = ncol(flash_fit$loadings$normalized.loadings[[1]]) 

# number of individuals
n = nrow(flash_fit$loadings$normalized.loadings[[1]])

# number of SNPs
p = nrow(flash_fit$loadings$normalized.loadings[[2]])
print(K)</code></pre>
<pre><code>[1] 21</code></pre>
<pre class="r"><code>print(n)</code></pre>
<pre><code>[1] 2068</code></pre>
<pre class="r"><code>print(p)</code></pre>
<pre><code>[1] 165468</code></pre>
</div>
<div id="pves" class="section level1">
<h1>PVEs</h1>
<p>Here is a plot of the proportion of variance (PVE) explained by each drift event:</p>
<pre class="r"><code>p_pve = qplot(2:K, flash_fit$pve[2:K]) + 
        ylab(&quot;Proportion of Varaince Explained&quot;) + 
        xlab(&quot;K&quot;) + 
        theme_bw()
p_pve</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-pve-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-greedy-ld-viz-pve-1">
Past versions of flash-greedy-ld-viz-pve-1.png
</button>
</p>
<div id="fig-flash-greedy-ld-viz-pve-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-pve-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-pve-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-pve-1.png" target="_blank">38b57c5</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>print(flash_fit$pve)</code></pre>
<pre><code> [1] 0.4758928191 0.0180430938 0.0221910986 0.0102953222 0.0025186497
 [6] 0.0045329056 0.0015651404 0.0014220784 0.0011824621 0.0008636509
[11] 0.0007593062 0.0006714457 0.0005026230 0.0002140494 0.0001931981
[16] 0.0002269361 0.0001832276 0.0002904277 0.0002626722 0.0001573182
[21] 0.0001179608</code></pre>
<p>It looks like the PVE drops off at around 14 or so?</p>
</div>
<div id="fitted-mean-and-variance" class="section level1">
<h1>Fitted Mean and Variance</h1>
<p>I setup the <code>flashier</code> run so it estimates a SNP specific precision term. Here is a histogram of fitted variances:</p>
<pre class="r"><code>p_var = qplot(1/flash_fit$fit$est.tau) + 
        xlab(&quot;Estimated Variance&quot;) + 
        ylab(&quot;Count&quot;) + 
        theme_bw()
p_var</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-var-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-greedy-ld-viz-var-1">
Past versions of flash-greedy-ld-viz-var-1.png
</button>
</p>
<div id="fig-flash-greedy-ld-viz-var-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-var-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-var-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-var-1.png" target="_blank">38b57c5</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Lets now look the the fitted means:</p>
<pre class="r"><code>mu = sqrt(flash_fit$loadings$scale.constant[1]) * delta_df$`1`
p_mu = qplot(mu) + 
       xlab(&quot;Estimated Mean&quot;) + 
       ylab(&quot;Count&quot;) + 
       theme_bw()
p_mu</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-mean-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-greedy-ld-viz-mean-1">
Past versions of flash-greedy-ld-viz-mean-1.png
</button>
</p>
<div id="fig-flash-greedy-ld-viz-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mean-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mean-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mean-1.png" target="_blank">38b57c5</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>These plots looks about reasonable as each of the SNP variances should roughly be interpreted as average heterozygosity <span class="math inline">\(\approx 2p(1-p)\)</span>? The mean term should roughly be interpreted as the mean minor allele frequency at the SNP and thus we should see a quadratic relationship with the estimated variance:</p>
<pre class="r"><code>d1 = flash_fit$loadings$scale.constant[1]
mv_df = data.frame(var=1/flash_fit$fit$est.tau, mu=mu, chrom=bim_df$chrom)
p_mv = ggplot(mv_df, aes(x=mu, y=var)) + 
       geom_point() + 
       xlab(&quot;Estimated Mean&quot;) + ylab(&quot;Estimated Variance&quot;) + 
       scale_alpha(guide = &quot;none&quot;) + 
       stat_function(fun = function(x){return(2*x*(1-x))}, color=&quot;red&quot;) + 
       xlim(0, .5) + 
       theme_bw()
p_mv</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-mv-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-greedy-ld-viz-mv-1">
Past versions of flash-greedy-ld-viz-mv-1.png
</button>
</p>
<div id="fig-flash-greedy-ld-viz-mv-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mv-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mv-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/63e7173f9a33d2a087e6f16d90cd929eb422bb6b/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mv-1.png" target="_blank">63e7173</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mv-1.png" target="_blank">38b57c5</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Most of the SNPs have a mean-variance relationship expected under a simple Binomial model for the genotypes i.e. <span class="math inline">\(y_{ij} \sim Binomial(2, p_{ij})\)</span>. I wonder if there is anything “special” going on with the high variance SNP (will explore this later). I’m not sure why there is a sharp cuttof at .45.</p>
</div>
<div id="drift-event-distributions" class="section level1">
<h1>Drift Event Distributions</h1>
<p>Lets now plot the distribution of drift events:</p>
<pre class="r"><code># gather the data.frame for plotting
delta_gath_df = delta_df %&gt;% 
                gather(K, value, -chrom, -pos, -rsid) %&gt;%
                filter(K!=1)

# plot the factors
K_ = K
p_de = ggplot(delta_gath_df, aes(x=value)) + 
        geom_histogram() + 
        facet_wrap(~factor(K, levels=2:K_), scales = &quot;free&quot;) + 
        labs(fill=&quot;K&quot;) + 
        scale_x_continuous(breaks = scales::pretty_breaks(n = 3)) +
        scale_y_continuous(breaks = scales::pretty_breaks(n = 3)) + 
        theme_bw()
p_de</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-factors-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-greedy-ld-viz-factors-1">
Past versions of flash-greedy-ld-viz-factors-1.png
</button>
</p>
<div id="fig-flash-greedy-ld-viz-factors-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-factors-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-factors-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-factors-1.png" target="_blank">38b57c5</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/f5ef1af4d17d9469ed52811c88166186d0c3c993/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-factors-1.png" target="_blank">f5ef1af</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can see the lower PVE drift events tend to get sparser!</p>
</div>
<div id="drift-event-local-false-sign-rates" class="section level1">
<h1>Drift Event Local False Sign Rates</h1>
<p>Here histograms of the lfsrs for each drift event:</p>
<pre class="r"><code>delta_lfsr_gath_df = delta_lfsr_df %&gt;% 
                     gather(K, value, -chrom, -pos, -rsid) %&gt;%
                     filter(K %in% paste0(2:21))

p_lfsr = ggplot(delta_lfsr_gath_df, aes(x=value)) + 
         geom_histogram() + 
         facet_wrap(~factor(K, levels=2:K_), scales = &quot;free&quot;) + 
         labs(fill=&quot;K&quot;) + 
         scale_x_continuous(breaks = scales::pretty_breaks(n = 3)) +
         scale_y_continuous(breaks = scales::pretty_breaks(n = 3)) + 
         theme_bw() + 
         xlab(&quot;Local False Sign Rate&quot;)
p_lfsr</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-lfsr-factors-viz-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-lfsr-factors-viz-1">
Past versions of flash-lfsr-factors-viz-1.png
</button>
</p>
<div id="fig-flash-lfsr-factors-viz-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-lfsr-factors-viz-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>It is interesting to see the lower PVE factors (around 13 onwards) shift to having many SNPs with high uncertainity of the sign of the drift event.</p>
</div>
<div id="drift-event-loadings-2-11" class="section level1">
<h1>Drift Event Loadings (2-11)</h1>
<p>Lets now take a look at the drift event loadings. First we setup the data:</p>
<pre class="r"><code># read the meta data
meta_df = read.table(&quot;../data/meta/HumanOriginsPublic2068.meta&quot;, sep=&quot;\t&quot;, header=T)

# setup loadings data.frame
l_df = as.data.frame(flash_fit$loadings$normalized.loadings[[1]])
K = ncol(l_df)
l_df = cbind(l_df, meta_df)

# all unique pop labels
pops = unique(l_df$Simple.Population.ID) 

# join with the meta data
l_df = l_df %&gt;% arrange(Region, Simple.Population.ID) # sort by region then by population
l_df$ID = factor(l_df$ID, levels = l_df$ID) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

head(l_df)</code></pre>
<pre><code>           1            2           3          4            5            6
1 0.02198997 1.562440e-10 0.009748510 0.02224623 7.496723e-07 5.564681e-07
2 0.02198997 1.498617e-10 0.009740473 0.02269808 1.389697e-06 4.160829e-07
3 0.02198997 1.450794e-10 0.006494890 0.02497461 4.110090e-07 1.022280e-06
4 0.02198997 1.467553e-10 0.009433912 0.02314688 4.900230e-07 7.339584e-07
5 0.02198997 1.483937e-10 0.009868851 0.02290795 1.023726e-06 1.243865e-06
6 0.02198997 1.557400e-10 0.009509020 0.02230823 4.839558e-07 9.636279e-07
            7            8            9         10           11
1 0.011864108 1.950983e-06 2.134680e-08 0.05518295 1.998480e-06
2 0.010896971 2.218510e-06 2.596953e-08 0.04932135 4.235657e-06
3 0.010882016 8.834097e-07 1.532842e-08 0.05498083 3.759938e-06
4 0.011011579 1.021461e-06 1.829201e-08 0.05974476 1.691153e-06
5 0.008335442 1.722933e-06 1.676397e-08 0.05778697 1.862318e-06
6 0.011669452 7.847559e-07 2.056824e-08 0.05614764 1.171816e-06
           12           13           14         15           16
1 0.014841288 2.475559e-06 1.196299e-06 0.06257501 3.574080e-06
2 0.009671195 1.735489e-05 1.307078e-06 0.04815442 5.286249e-06
3 0.011501083 8.012301e-06 1.163721e-06 0.06577732 4.371689e-06
4 0.016477949 4.847320e-06 6.057587e-07 0.07252450 6.484941e-06
5 0.016029848 1.854048e-06 7.768784e-07 0.07805249 5.937048e-06
6 0.015462082 3.375407e-06 1.009148e-06 0.06144209 4.287258e-06
            17           18           19           20           21
1 3.231384e-05 3.686193e-04 2.768013e-05 1.201701e-05 2.821169e-06
2 4.278782e-05 2.089317e-04 4.289490e-04 4.107142e-05 2.684191e-06
3 2.555584e-03 2.272919e-04 3.728350e-05 1.835738e-04 2.204598e-06
4 5.754635e-05 5.062869e-05 2.207929e-05 7.701641e-04 1.574267e-06
5 2.230132e-05 8.553066e-05 1.794721e-04 7.282639e-03 2.575973e-06
6 5.395636e-05 5.777995e-04 1.192646e-05 1.745599e-05 2.016293e-06
             ID Simple.Population.ID Verbose.Population.ID Region Country
1 Algerian43A22             Algerian              Algerian Africa Algeria
2 Algerian43A21             Algerian              Algerian Africa Algeria
3 Algerian43A34             Algerian              Algerian Africa Algeria
4 Algerian43A13             Algerian              Algerian Africa Algeria
5 Algerian43A24             Algerian              Algerian Africa Algeria
6 Algerian43A32             Algerian              Algerian Africa Algeria
  Latitude Longitude Samples Passed.QC Contributor
1     36.8         3       7         7 David Comas
2     36.8         3       7         7 David Comas
3     36.8         3       7         7 David Comas
4     36.8         3       7         7 David Comas
5     36.8         3       7         7 David Comas
6     36.8         3       7         7 David Comas</code></pre>
<p>Its hard to find a color scale that can sufficiently visualize all of the loadings in a single plot. Instead I just split the loadings up into two plots (K=2,…,11) and (K=12,…,21). Lets first visualize loadings 2 through 12:</p>
<pre class="r"><code># gather the data.frame for plotting
l_gath_df = l_df %&gt;% 
            gather(K, value, 
                   -ID,
                   -Verbose.Population.ID, 
                   -Simple.Population.ID, 
                   -Region, -Country,
                   -Latitude,
                   -Longitude,
                   -Samples,
                   -Passed.QC,
                   -Contributor) %&gt;% 
            filter(K %in% paste0(2:11))

# Africa
africa_pops = get_pops(meta_df, &quot;Africa&quot;)
p_africa = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                   filter(Region == &quot;Africa&quot;), 
                                   colset=&quot;Set3&quot;,
                                   facet_levels=africa_pops,
                                   facet_grp=&quot;Simple.Population.ID&quot;, 
                                   label_size=5) +
           ggtitle(&quot;Africa&quot;) + 
           theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, &quot;America&quot;)
p_america = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                    filter(Region == &quot;America&quot;), 
                                    colset=&quot;Set3&quot;,
                                    facet_levels=america_pops,
                                    facet_grp=&quot;Simple.Population.ID&quot;, 
                                    label_size=5) + 
            ggtitle(&quot;America&quot;) + 
            theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, &quot;CentralAsiaSiberia&quot;)
p_central_asia_siberia = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                                 filter(Region == &quot;CentralAsiaSiberia&quot;), 
                                                 colset=&quot;Set3&quot;,
                                                 facet_levels=central_asia_siberia_pops,  
                                                 facet_grp=&quot;Simple.Population.ID&quot;,
                                                 label_size=5) + 
                         ggtitle(&quot;CentralAsiaSiberia&quot;) + 
                         theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, &quot;EastAsia&quot;)
p_east_asia = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                      filter(Region == &quot;EastAsia&quot;), 
                                      colset=&quot;Set3&quot;,
                                      facet_levels=east_asia_pops,  
                                      facet_grp=&quot;Simple.Population.ID&quot;,
                                      label_size=5) + 
              ggtitle(&quot;EastAsia&quot;) + 
              theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, &quot;SouthAsia&quot;)
p_south_asia= positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                      filter(Region == &quot;SouthAsia&quot;),
                                      colset=&quot;Set3&quot;,
                                      facet_levels=south_asia_pops, 
                                      facet_grp=&quot;Simple.Population.ID&quot;,
                                      label_size=5) + 
              ggtitle(&quot;SouthAsia&quot;) + 
              theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, &quot;WestEurasia&quot;)
p_west_eurasia = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                         filter(Region == &quot;WestEurasia&quot;), 
                                         colset=&quot;Set3&quot;,
                                         facet_levels=west_eurasia_pops, 
                                         facet_grp=&quot;Simple.Population.ID&quot;,
                                         label_size=5) + 
                 ggtitle(&quot;WestEurasia&quot;) + 
                 theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, &quot;Oceania&quot;)
p_oceania = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                    filter(Region == &quot;Oceania&quot;), 
                                    colset=&quot;Set3&quot;,
                                    facet_levels=oceania_pops, 
                                    facet_grp=&quot;Simple.Population.ID&quot;,
                                    label_size=5) + 
            ggtitle(&quot;Oceania&quot;) + 
            theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia,
                       p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = &quot;v&quot;) 
p</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-2-11-1.png" width="816" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-greedy-ld-viz-loadings-2-11-1">
Past versions of flash-greedy-ld-viz-loadings-2-11-1.png
</button>
</p>
<div id="fig-flash-greedy-ld-viz-loadings-2-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-2-11-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-2-11-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-2-11-1.png" target="_blank">38b57c5</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="drift-event-loadings-12-21" class="section level1">
<h1>Drift Event Loadings (12-21)</h1>
<p>Lets now visualize loadings 12 to 21 (be careful: there is no connection to the colors in the last plot):</p>
<pre class="r"><code># gather the data.frame for plotting
l_gath_df = l_df %&gt;% 
            gather(K, value, 
                   -ID,
                   -Verbose.Population.ID, 
                   -Simple.Population.ID, 
                   -Region, -Country,
                   -Latitude,
                   -Longitude,
                   -Samples,
                   -Passed.QC,
                   -Contributor) %&gt;% 
            filter(K %in% paste0(12:21))

# Africa
africa_pops = get_pops(meta_df, &quot;Africa&quot;)
p_africa = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                   filter(Region == &quot;Africa&quot;), 
                                   colset=&quot;Set3&quot;,
                                   facet_levels=africa_pops,
                                   facet_grp=&quot;Simple.Population.ID&quot;, 
                                   label_size=5) +
           ggtitle(&quot;Africa&quot;) + 
           theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, &quot;America&quot;)
p_america = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                    filter(Region == &quot;America&quot;), 
                                    colset=&quot;Set3&quot;,
                                    facet_levels=america_pops,
                                    facet_grp=&quot;Simple.Population.ID&quot;, 
                                    label_size=5) + 
            ggtitle(&quot;America&quot;) + 
            theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, &quot;CentralAsiaSiberia&quot;)
p_central_asia_siberia = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                                 filter(Region == &quot;CentralAsiaSiberia&quot;), 
                                                 colset=&quot;Set3&quot;,
                                                 facet_levels=central_asia_siberia_pops,  
                                                 facet_grp=&quot;Simple.Population.ID&quot;,
                                                 label_size=5) + 
                         ggtitle(&quot;CentralAsiaSiberia&quot;) + 
                         theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, &quot;EastAsia&quot;)
p_east_asia = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                      filter(Region == &quot;EastAsia&quot;), 
                                      colset=&quot;Set3&quot;,
                                      facet_levels=east_asia_pops,  
                                      facet_grp=&quot;Simple.Population.ID&quot;,
                                      label_size=5) + 
              ggtitle(&quot;EastAsia&quot;) + 
              theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, &quot;SouthAsia&quot;)
p_south_asia= positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                      filter(Region == &quot;SouthAsia&quot;),
                                      colset=&quot;Set3&quot;,
                                      facet_levels=south_asia_pops, 
                                      facet_grp=&quot;Simple.Population.ID&quot;,
                                      label_size=5) + 
              ggtitle(&quot;SouthAsia&quot;) + 
              theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, &quot;WestEurasia&quot;)
p_west_eurasia = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                         filter(Region == &quot;WestEurasia&quot;), 
                                         colset=&quot;Set3&quot;,
                                         facet_levels=west_eurasia_pops, 
                                         facet_grp=&quot;Simple.Population.ID&quot;,
                                         label_size=5) + 
                 ggtitle(&quot;WestEurasia&quot;) + 
                 theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, &quot;Oceania&quot;)
p_oceania = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                    filter(Region == &quot;Oceania&quot;), 
                                    colset=&quot;Set3&quot;,
                                    facet_levels=oceania_pops, 
                                    facet_grp=&quot;Simple.Population.ID&quot;,
                                    label_size=5) + 
            ggtitle(&quot;Oceania&quot;) + 
            theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia,
                       p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = &quot;v&quot;) 
p</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-12-21-1.png" width="816" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-greedy-ld-viz-loadings-12-21-1">
Past versions of flash-greedy-ld-viz-loadings-12-21-1.png
</button>
</p>
<div id="fig-flash-greedy-ld-viz-loadings-12-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-12-21-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-12-21-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-12-21-1.png" target="_blank">38b57c5</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Its kinda interesting to see that some populations have zero loading on later factors. Its also interesting to see a lot of population specific factors arising. This would be difficult to visualize see if using a single plot for all the factors.</p>
</div>
<div id="drift-event-loadings-local-false-sign-rates" class="section level1">
<h1>Drift Event Loadings Local False Sign Rates</h1>
<pre class="r"><code>l_lfsr_gath_df = l_lfsr_df %&gt;% 
                 gather(K, value) %&gt;%
                 filter(K %in% paste0(2:21))

p_lfsr = ggplot(l_lfsr_gath_df, aes(x=value)) + 
         geom_histogram() + 
         facet_wrap(~factor(K, levels=2:K_), scales = &quot;free&quot;) + 
         labs(fill=&quot;K&quot;) + 
         scale_x_continuous(breaks = scales::pretty_breaks(n = 3)) +
         scale_y_continuous(breaks = scales::pretty_breaks(n = 3)) + 
         theme_bw() + 
         xlab(&quot;Local False Sign Rate&quot;)
p_lfsr</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-lfsr-loadings-viz-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-lfsr-loadings-viz-1">
Past versions of flash-lfsr-loadings-viz-1.png
</button>
</p>
<div id="fig-flash-lfsr-loadings-viz-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-lfsr-loadings-viz-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Its very intersting to see that the lfsr seem very bi-model for the loadings for many of the drift events.</p>
</div>
<div id="admixture" class="section level1">
<h1>ADMIXTURE</h1>
<p>Lets visualize ADMIXTURE with 9 factors which should roughly align to the first plot i.e. <code>FLASH</code> with 2,…,11 (be careful: there is no connection to the colors in the last plot):</p>
<pre class="r"><code>l_df = read.table(&quot;../output/admixture/hoa_global_ld/HumanOriginsPublic2068_maf_geno_auto_ldprune.K9r1.Q&quot;, sep=&quot; &quot;, header=F)
K = ncol(l_df)
l_df = cbind(l_df, meta_df)
pops = unique(l_df$Simple.Population.ID) # all unique pop labels
l_df = l_df %&gt;% arrange(Region, Simple.Population.ID) # sort by region then by population
l_df$ID = factor(l_df$ID, levels = l_df$ID) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

# gather the data.frame for plotting
l_gath_df = l_df %&gt;% 
            gather(K, value, -ID, -Verbose.Population.ID, -Simple.Population.ID, 
                   -Region, -Country, -Latitude, -Longitude, -Samples,
                   -Passed.QC,  -Contributor) 

# Africa
africa_pops = get_pops(meta_df, &quot;Africa&quot;)
p_africa = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                   filter(Region == &quot;Africa&quot;), 
                                   colset=&quot;Set3&quot;,
                                   facet_levels=africa_pops,
                                   facet_grp=&quot;Simple.Population.ID&quot;, 
                                   label_size=5) +
           ggtitle(&quot;Africa&quot;) + 
           theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, &quot;America&quot;)
p_america = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                    filter(Region == &quot;America&quot;), 
                                    colset=&quot;Set3&quot;,
                                    facet_levels=america_pops,
                                    facet_grp=&quot;Simple.Population.ID&quot;, 
                                    label_size=5) + 
            ggtitle(&quot;America&quot;) + 
            theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, &quot;CentralAsiaSiberia&quot;)
p_central_asia_siberia = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                                 filter(Region == &quot;CentralAsiaSiberia&quot;), 
                                                 colset=&quot;Set3&quot;,
                                                 facet_levels=central_asia_siberia_pops,  
                                                 facet_grp=&quot;Simple.Population.ID&quot;,
                                                 label_size=5) + 
                         ggtitle(&quot;CentralAsiaSiberia&quot;) + 
                         theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, &quot;EastAsia&quot;)
p_east_asia = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                      filter(Region == &quot;EastAsia&quot;), 
                                      colset=&quot;Set3&quot;,
                                      facet_levels=east_asia_pops,  
                                      facet_grp=&quot;Simple.Population.ID&quot;,
                                      label_size=5) + 
              ggtitle(&quot;EastAsia&quot;) + 
              theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, &quot;SouthAsia&quot;)
p_south_asia= positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                      filter(Region == &quot;SouthAsia&quot;),
                                      colset=&quot;Set3&quot;,
                                      facet_levels=south_asia_pops, 
                                      facet_grp=&quot;Simple.Population.ID&quot;,
                                      label_size=5) + 
              ggtitle(&quot;SouthAsia&quot;) + 
              theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, &quot;WestEurasia&quot;)
p_west_eurasia = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                         filter(Region == &quot;WestEurasia&quot;), 
                                         colset=&quot;Set3&quot;,
                                         facet_levels=west_eurasia_pops, 
                                         facet_grp=&quot;Simple.Population.ID&quot;,
                                         label_size=5) + 
                 ggtitle(&quot;WestEurasia&quot;) + 
                 theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, &quot;Oceania&quot;)
p_oceania = positive_structure_plot(gath_df=l_gath_df %&gt;% 
                                    filter(Region == &quot;Oceania&quot;), 
                                    colset=&quot;Set3&quot;,
                                    facet_levels=oceania_pops, 
                                    facet_grp=&quot;Simple.Population.ID&quot;,
                                    label_size=5) + 
            ggtitle(&quot;Oceania&quot;) + 
            theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia,
                       p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = &quot;v&quot;) 
p</code></pre>
<p><img src="figure/hoa_global.Rmd/admixture-ld-viz-loadings-9-1.png" width="816" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-admixture-ld-viz-loadings-9-1">
Past versions of admixture-ld-viz-loadings-9-1.png
</button>
</p>
<div id="fig-admixture-ld-viz-loadings-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/hoa_global.Rmd/admixture-ld-viz-loadings-9-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>There is a lot that one can compare between the ADMIXTURE and FLASH results. A high level observation seems that the ADMIXTURE results look a bit more clustered i.e. the Americas and East Asia look like they are explained mostly by 1 or 2 factors whereas FLASH uses 3-4. Its hard to tell be it seems that this is true in many of the super regions … ADMIXTURE tends use fewer factors to explain population structure in each region, leading to a more clustered result?</p>
</div>
<div id="outlier-snps" class="section level1">
<h1>Outlier SNPs</h1>
<p>Lets take a closer look at the drift factors to see if they are clustering in particular regions of the genome. As a first pass I take the top 5% of SNPs weighted on each drift event (to be clear I ignore the sign of each SNP). I would like to use the lfsr here but I will return to later. I then made a Manhatten plot for each chromosome and factor:</p>
<pre class="r"><code>manh_df = delta_lfsr_df %&gt;% 
          group_by(chrom) %&gt;% 
          summarise(chr_len=max(pos)) %&gt;% 
          mutate(tot=cumsum(chr_len)-chr_len) %&gt;%
          dplyr::select(-chr_len) %&gt;%
          left_join(delta_lfsr_df, ., by=c(&quot;chrom&quot;=&quot;chrom&quot;)) %&gt;%
          arrange(chrom, pos) %&gt;%
          mutate(BPcum=pos+tot)

manh_axis_df = manh_df %&gt;% 
               group_by(chrom) %&gt;% 
               summarize(center=(max(BPcum) + min(BPcum)) / 2)

manh_gath_df = manh_df %&gt;% gather(K, value, -chrom, -pos, -rsid, -tot, -BPcum) %&gt;% 
               filter(K %in% paste0(2:11)) %&gt;% 
               filter(value &lt; 1e-10) %&gt;%
               filter(chrom %in% 1:22)
          
p = ggplot(manh_gath_df, aes(x=BPcum, y=-log10(value))) +
    geom_point(aes(color=as.factor(chrom)), alpha=.7, size=.5) +
    scale_color_manual(values=rep(c(&quot;grey&quot;, &quot;orange&quot;), 22)) +
    scale_x_continuous(label=manh_axis_df$chrom, breaks=manh_axis_df$center) +
    scale_y_continuous(expand=c(0, 0)) +     
    theme_bw() +
    theme(axis.title.x=element_blank(), axis.ticks.x=element_blank(),
          panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
    facet_grid(factor(K, levels=2:11)~., scales = &quot;free&quot;) + 
    ylab(&quot;-log10(lfsr)&quot;) + 
    guides(alpha=F, color=F)
p</code></pre>
<pre><code>Warning in FUN(X[[i]], ...): NaNs produced</code></pre>
<pre><code>Warning: Removed 1344 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-ld-factors-chrom-viz-1.png" width="816" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-flash-ld-factors-chrom-viz-1">
Past versions of flash-ld-factors-chrom-viz-1.png
</button>
</p>
<div id="fig-flash-ld-factors-chrom-viz-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/c203e00bbb6ad84d71308321e6e3b9e609c2ae21/docs/figure/hoa_global.Rmd/flash-ld-factors-chrom-viz-1.png" target="_blank">c203e00</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jhmarcus/drift-workflow/blob/90a0a0229a2e71016ddb9c88e8fd607ce92a567a/docs/figure/hoa_global.Rmd/flash-ld-factors-chrom-viz-1.png" target="_blank">90a0a02</a>
</td>
<td>
jhmarcus
</td>
<td>
2019-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can see there are some regions on the chromosomes that are peaky as well as some regions that have no “outliers” at all. I then took the top 5 outliers in each factor and annotated them with some functional information:</p>
<pre class="r"><code># SNP bioMart database
grch37_snp = useMart(biomart=&quot;ENSEMBL_MART_SNP&quot;, 
                     host=&quot;grch37.ensembl.org&quot;, 
                     path=&quot;/biomart/martservice&quot;,
                     dataset=&quot;hsapiens_snp&quot;)

# GENE bioMart database
grch37 = useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, 
                 host=&quot;grch37.ensembl.org&quot;, 
                 path=&quot;/biomart/martservice&quot;, 
                 dataset=&quot;hsapiens_gene_ensembl&quot;)

# top rank SNPs per each drift event
delta_tophit_df = delta_df %&gt;%
                  gather(K, value, -chrom, -pos, -rsid) %&gt;%
                  filter(K %in% paste0(2:21)) %&gt;% 
                  group_by(K) %&gt;%
                  top_n(5, value)

# SNP meta data
table1 = getBM(attributes=c(&quot;refsnp_id&quot;, 
                             &quot;chrom_start&quot;, 
                             &quot;minor_allele_freq&quot;,
                             &quot;ensembl_gene_stable_id&quot;,
                             &quot;consequence_type_tv&quot;, 
                             &quot;associated_gene&quot;), 
                filters = &quot;snp_filter&quot;, 
                values = delta_tophit_df$rsid, 
                mart = grch37_snp)
table1$ensembl_gene_id = table1$ensembl_gene_stable_id

# GENE meta data
table2 = getBM(attributes = c(&quot;ensembl_gene_id&quot;, &quot;external_gene_name&quot;, &quot;description&quot;),
               filters = &quot;ensembl_gene_id&quot;, 
               values =  table1$ensembl_gene_stable_id, 
               mart = grch37)

# annotation data
anno_df = table1 %&gt;% left_join(table2, on=&quot;ensembl_gene_id&quot;) %&gt;% 
          mutate(rsid=refsnp_id) %&gt;% 
          inner_join(delta_tophit_df, on=&quot;rsid&quot;) </code></pre>
<pre><code>Joining, by = &quot;ensembl_gene_id&quot;</code></pre>
<pre><code>Joining, by = &quot;rsid&quot;</code></pre>
<pre><code>Warning: Column `rsid` joining character vector and factor, coercing into
character vector</code></pre>
<pre class="r"><code># unique genes
print(unique(anno_df$external_gene_name))</code></pre>
<pre><code> [1] &quot;CCNL2&quot;          &quot;CRYZ&quot;           &quot;SLC22A15&quot;       NA              
 [5] &quot;RP11-280O1.2&quot;   &quot;KIAA1614&quot;       &quot;C1orf132&quot;       &quot;CD34&quot;          
 [9] &quot;FRMD4A&quot;         &quot;MPP7&quot;           &quot;SGPL1&quot;          &quot;CCSER2&quot;        
[13] &quot;RBM20&quot;          &quot;NAV2&quot;           &quot;SLC22A10&quot;       &quot;STT3A-AS1&quot;     
[17] &quot;STT3A&quot;          &quot;HOXC13-AS&quot;      &quot;CNOT2&quot;          &quot;RP11-114H23.1&quot; 
[21] &quot;APPL2&quot;          &quot;LINC00641&quot;      &quot;HERC2&quot;          &quot;RP11-109D20.1&quot; 
[25] &quot;SORD&quot;           &quot;RP11-266O8.1&quot;   &quot;PRKCB&quot;          &quot;PKD1L2&quot;        
[29] &quot;RP11-1407O15.2&quot; &quot;CACNB1&quot;         &quot;RP11-515E23.1&quot;  &quot;RP11-1055B8.4&quot; 
[33] &quot;LINC01119&quot;      &quot;IL1R1&quot;          &quot;RAB3GAP1&quot;       &quot;ZRANB3&quot;        
[37] &quot;ITGB6&quot;          &quot;TLK1&quot;           &quot;PARD3B&quot;         &quot;SF3A1&quot;         
[41] &quot;DRG1&quot;           &quot;TEF&quot;            &quot;KAT2B&quot;          &quot;SEMA3F&quot;        
[45] &quot;FRMD4B&quot;         &quot;MYH15&quot;          &quot;CRMP1&quot;          &quot;EVC&quot;           
[49] &quot;RP11-103J17.1&quot;  &quot;RP11-696N14.1&quot;  &quot;SLC45A2&quot;        &quot;NDUFAF2&quot;       
[53] &quot;CTB-35F21.1&quot;    &quot;DPYSL3&quot;         &quot;UNC5A&quot;          &quot;FLOT1&quot;         
[57] &quot;AKAP12&quot;         &quot;TIAM2&quot;          &quot;TMEM248&quot;        &quot;SMKR1&quot;         
[61] &quot;RP11-150O12.1&quot;  &quot;SMARCA2&quot;        &quot;GLIS3&quot;          &quot;RCL1&quot;          
[65] &quot;PTPN3&quot;         </code></pre>
<pre class="r"><code># formatted table
d = anno_df %&gt;% 
    distinct(external_gene_name, .keep_all = T) %&gt;% 
    dplyr::select(external_gene_name, K, rsid, consequence_type_tv) %&gt;% 
    arrange(consequence_type_tv)
kable(d)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">external_gene_name</th>
<th align="left">K</th>
<th align="left">rsid</th>
<th align="left">consequence_type_tv</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
<td align="left">3</td>
<td align="left">rs6428891</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">CRYZ</td>
<td align="left">11</td>
<td align="left">rs10890141</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">SLC22A15</td>
<td align="left">16</td>
<td align="left">rs17035177</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">KIAA1614</td>
<td align="left">15</td>
<td align="left">rs7545466</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">FRMD4A</td>
<td align="left">8</td>
<td align="left">rs201874034</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">MPP7</td>
<td align="left">4</td>
<td align="left">rs10826403</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">SGPL1</td>
<td align="left">8</td>
<td align="left">rs7083883</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">CCSER2</td>
<td align="left">21</td>
<td align="left">rs56347056</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">RBM20</td>
<td align="left">19</td>
<td align="left">rs11598929</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">NAV2</td>
<td align="left">3</td>
<td align="left">rs10741783</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">CNOT2</td>
<td align="left">7</td>
<td align="left">rs11178168</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">HERC2</td>
<td align="left">12</td>
<td align="left">rs7494942</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">SORD</td>
<td align="left">5</td>
<td align="left">rs2470686</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">PRKCB</td>
<td align="left">13</td>
<td align="left">rs4787651</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">CACNB1</td>
<td align="left">15</td>
<td align="left">rs16531</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">LINC01119</td>
<td align="left">13</td>
<td align="left">rs6713911</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">IL1R1</td>
<td align="left">13</td>
<td align="left">rs3755295</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">RAB3GAP1</td>
<td align="left">12</td>
<td align="left">rs6730157</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">ITGB6</td>
<td align="left">8</td>
<td align="left">rs12987602</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">TLK1</td>
<td align="left">11</td>
<td align="left">rs78813632</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">PARD3B</td>
<td align="left">10</td>
<td align="left">rs1207425</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">SF3A1</td>
<td align="left">5</td>
<td align="left">rs5749071</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">DRG1</td>
<td align="left">16</td>
<td align="left">rs2273248</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">TEF</td>
<td align="left">6</td>
<td align="left">rs9611566</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">KAT2B</td>
<td align="left">14</td>
<td align="left">rs2929402</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">SEMA3F</td>
<td align="left">9</td>
<td align="left">rs11717349</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">FRMD4B</td>
<td align="left">13</td>
<td align="left">rs34266487</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">MYH15</td>
<td align="left">2</td>
<td align="left">rs6437783</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">EVC</td>
<td align="left">9</td>
<td align="left">rs4689306</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">SLC45A2</td>
<td align="left">4</td>
<td align="left">rs185146</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">NDUFAF2</td>
<td align="left">11</td>
<td align="left">rs1841504</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">DPYSL3</td>
<td align="left">18</td>
<td align="left">rs17106725</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">UNC5A</td>
<td align="left">3</td>
<td align="left">rs692713</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">AKAP12</td>
<td align="left">16</td>
<td align="left">rs17081009</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">TIAM2</td>
<td align="left">6</td>
<td align="left">rs6916552</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">TMEM248</td>
<td align="left">8</td>
<td align="left">rs10264873</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">SMKR1</td>
<td align="left">14</td>
<td align="left">rs11771549</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">SMARCA2</td>
<td align="left">20</td>
<td align="left">rs7035991</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">GLIS3</td>
<td align="left">14</td>
<td align="left">rs10974315</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">RCL1</td>
<td align="left">3</td>
<td align="left">rs172447</td>
<td align="left">intron_variant</td>
</tr>
<tr class="odd">
<td align="left">PTPN3</td>
<td align="left">18</td>
<td align="left">rs12336101</td>
<td align="left">intron_variant</td>
</tr>
<tr class="even">
<td align="left">CCNL2</td>
<td align="left">4</td>
<td align="left">rs1240747</td>
<td align="left">NMD_transcript_variant</td>
</tr>
<tr class="odd">
<td align="left">RP11-1407O15.2</td>
<td align="left">10</td>
<td align="left">rs8066255</td>
<td align="left">NMD_transcript_variant</td>
</tr>
<tr class="even">
<td align="left">C1orf132</td>
<td align="left">21</td>
<td align="left">rs2235767</td>
<td align="left">non_coding_transcript_exon_variant</td>
</tr>
<tr class="odd">
<td align="left">LINC00641</td>
<td align="left">6</td>
<td align="left">rs12184962</td>
<td align="left">non_coding_transcript_exon_variant</td>
</tr>
<tr class="even">
<td align="left">PKD1L2</td>
<td align="left">19</td>
<td align="left">rs16954698</td>
<td align="left">non_coding_transcript_exon_variant</td>
</tr>
<tr class="odd">
<td align="left">RP11-1055B8.4</td>
<td align="left">5</td>
<td align="left">rs3744146</td>
<td align="left">non_coding_transcript_exon_variant</td>
</tr>
<tr class="even">
<td align="left">RP11-280O1.2</td>
<td align="left">2</td>
<td align="left">rs4657449</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="odd">
<td align="left">CD34</td>
<td align="left">21</td>
<td align="left">rs3820521</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="even">
<td align="left">SLC22A10</td>
<td align="left">13</td>
<td align="left">rs6591765</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="odd">
<td align="left">STT3A-AS1</td>
<td align="left">20</td>
<td align="left">rs503288</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="even">
<td align="left">HOXC13-AS</td>
<td align="left">5</td>
<td align="left">rs2366148</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="odd">
<td align="left">RP11-114H23.1</td>
<td align="left">14</td>
<td align="left">rs3847673</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="even">
<td align="left">RP11-109D20.1</td>
<td align="left">5</td>
<td align="left">rs2470686</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="odd">
<td align="left">RP11-266O8.1</td>
<td align="left">4</td>
<td align="left">rs7163903</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="even">
<td align="left">RP11-515E23.1</td>
<td align="left">5</td>
<td align="left">rs9908046</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="odd">
<td align="left">ZRANB3</td>
<td align="left">12</td>
<td align="left">rs6730157</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="even">
<td align="left">CRMP1</td>
<td align="left">9</td>
<td align="left">rs4689306</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="odd">
<td align="left">RP11-103J17.1</td>
<td align="left">9</td>
<td align="left">rs11733992</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="even">
<td align="left">RP11-696N14.1</td>
<td align="left">7</td>
<td align="left">rs1229966</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="odd">
<td align="left">CTB-35F21.1</td>
<td align="left">15</td>
<td align="left">rs10067965</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="even">
<td align="left">FLOT1</td>
<td align="left">15</td>
<td align="left">rs3094127</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="odd">
<td align="left">RP11-150O12.1</td>
<td align="left">11</td>
<td align="left">rs2407278</td>
<td align="left">non_coding_transcript_variant</td>
</tr>
<tr class="even">
<td align="left">STT3A</td>
<td align="left">20</td>
<td align="left">rs503288</td>
<td align="left">splice_region_variant</td>
</tr>
<tr class="odd">
<td align="left">APPL2</td>
<td align="left">20</td>
<td align="left">rs12303948</td>
<td align="left">synonymous_variant</td>
</tr>
</tbody>
</table>
We can see many of these top outliers are in genes (but we have to consider the array design to know if this is unusual). Its cool to see a couple very interesting genes pop up including HERC2 (eye color) and SLC45A2 (skin color) which have been previously studied for their selection signatures. Its also fun to look at some of the <code>rsids</code> on <code>https://popgen.uchicago.edu/ggv/</code> to get a sense of what kind of allele frequency distributions define each factor. <br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS  10.14.2

Matrix products: default
BLAS/LAPACK: /Users/jhmarcus/miniconda3/lib/R/lib/libRblas.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.21         biomaRt_2.38.0     RColorBrewer_1.1-2
[4] dplyr_0.8.0.1      tidyr_0.8.2        ggplot2_3.1.0     

loaded via a namespace (and not attached):
 [1] progress_1.2.0       tidyselect_0.2.5     xfun_0.4            
 [4] reshape2_1.4.3       purrr_0.3.0          colorspace_1.4-0    
 [7] htmltools_0.3.6      stats4_3.5.1         yaml_2.2.0          
[10] blob_1.1.1           XML_3.98-1.12        rlang_0.3.1         
[13] pillar_1.3.1         glue_1.3.0           withr_2.1.2         
[16] DBI_1.0.0            BiocGenerics_0.28.0  bit64_0.9-7         
[19] plyr_1.8.4           stringr_1.4.0        munsell_0.5.0       
[22] gtable_0.2.0         workflowr_1.2.0      flashier_0.1.0      
[25] evaluate_0.12        memoise_1.1.0        labeling_0.3        
[28] Biobase_2.42.0       IRanges_2.16.0       curl_3.3            
[31] parallel_3.5.1       AnnotationDbi_1.44.0 highr_0.7           
[34] Rcpp_1.0.0           scales_1.0.0         backports_1.1.3     
[37] S4Vectors_0.20.1     fs_1.2.6             bit_1.1-14          
[40] hms_0.4.2            digest_0.6.18        stringi_1.2.4       
[43] cowplot_0.9.4        grid_3.5.1           rprojroot_1.3-2     
[46] tools_3.5.1          bitops_1.0-6         magrittr_1.5        
[49] lazyeval_0.2.1       RCurl_1.95-4.11      tibble_2.0.1        
[52] RSQLite_2.1.1        crayon_1.3.4         whisker_0.3-2       
[55] pkgconfig_2.0.2      prettyunits_1.0.2    assertthat_0.2.0    
[58] rmarkdown_1.11       httr_1.4.0           R6_2.4.0            
[61] git2r_0.23.0         compiler_3.5.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
