<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="jhmarcus" />

<meta name="date" content="2019-02-15" />

<title>Human Origins Array Global Results</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">drift-workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Human Origins Array Global Results</h1>
<h4 class="author"><em>jhmarcus</em></h4>
<h4 class="date"><em>2019-02-15</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-02-24</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20190211)</code> </summary></p>
<p>The command <code>set.seed(20190211)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/jhmarcus/drift-workflow/tree/63e7173f9a33d2a087e6f16d90cd929eb422bb6b" target="_blank">63e7173</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    output.tar.gz
    Ignored:    output/flash_backfit/
    Ignored:    output/flash_greedy/
    Ignored:    output/log/
    Ignored:    output/sim/
    Ignored:    output/softImpute/

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/63e7173f9a33d2a087e6f16d90cd929eb422bb6b/analysis/hoa_global.Rmd" target="_blank">63e7173</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
<td style="text-align:left;">
expanded upon mean variance
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jhmarcus/drift-workflow/63e7173f9a33d2a087e6f16d90cd929eb422bb6b/docs/hoa_global.html" target="_blank">63e7173</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
<td style="text-align:left;">
expanded upon mean variance
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/analysis/hoa_global.Rmd" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
<td style="text-align:left;">
simplified greedy flash global analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jhmarcus/drift-workflow/38b57c54997e44c495298de7423104f1749299a1/docs/hoa_global.html" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
<td style="text-align:left;">
simplified greedy flash global analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/403bc6becb8b3a3bf9c901aa0cb3b9df258ac14e/analysis/hoa_global.Rmd" target="_blank">403bc6b</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
<td style="text-align:left;">
added hide code blocks
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jhmarcus/drift-workflow/403bc6becb8b3a3bf9c901aa0cb3b9df258ac14e/docs/hoa_global.html" target="_blank">403bc6b</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
<td style="text-align:left;">
added hide code blocks
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/b4749ac6e4cb22e7da9e07c2f03a1baacbf48cb2/analysis/hoa_global.Rmd" target="_blank">b4749ac</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
<td style="text-align:left;">
fixed some typos
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jhmarcus/drift-workflow/b4749ac6e4cb22e7da9e07c2f03a1baacbf48cb2/docs/hoa_global.html" target="_blank">b4749ac</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
<td style="text-align:left;">
fixed some typos
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/7a2b6c7d6a6827a391b617d74eb09098b1873a9a/analysis/hoa_global.Rmd" target="_blank">7a2b6c7</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
<td style="text-align:left;">
added backfit
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jhmarcus/drift-workflow/7a2b6c7d6a6827a391b617d74eb09098b1873a9a/docs/hoa_global.html" target="_blank">7a2b6c7</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
<td style="text-align:left;">
added backfit
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/f5ef1af4d17d9469ed52811c88166186d0c3c993/analysis/hoa_global.Rmd" target="_blank">f5ef1af</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
<td style="text-align:left;">
added workflows for human origins datasets
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jhmarcus/drift-workflow/f5ef1af4d17d9469ed52811c88166186d0c3c993/docs/hoa_global.html" target="_blank">f5ef1af</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
<td style="text-align:left;">
added workflows for human origins datasets
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/4afc77e7c21b668a5b240c920c563dab67acc0e5/analysis/hoa_global.Rmd" target="_blank">4afc77e</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
<td style="text-align:left;">
init hoa global analysis
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<div id="imports" class="section level2">
<h2>Imports</h2>
<p>Lets import some needed packages:</p>
<pre class="r"><code>library(ggplot2)
library(tidyr)
library(dplyr)
library(RColorBrewer)
source(&quot;../code/viz.R&quot;)</code></pre>
</div>
<div id="human-origins-global-ld-pruned" class="section level2">
<h2>Human Origins Global (LD Pruned)</h2>
<p>This is the full Human Origins dataset 2068 sampled from around the world. I filtered out rare variants with global minor allele frequency less than 5%, and remove any variants with a missingness level greater than 1%. I then LD pruned the SNPs using standard parameters in <code>plink</code>, resulting in 167178 SNPs.</p>
<div id="greedy" class="section level3">
<h3>Greedy</h3>
<p>Lets first read the greedy <code>flashier</code> fit</p>
<pre class="r"><code>flash_fit = readRDS(&quot;../output/flash_greedy/hoa_global_ld/HumanOriginsPublic2068_maf_geno_ldprune.rds&quot;)
K = ncol(flash_fit$loadings$normalized.loadings[[1]]) 
n = nrow(flash_fit$loadings$normalized.loadings[[1]])
p = nrow(flash_fit$loadings$normalized.loadings[[2]])
print(K)</code></pre>
<pre><code>[1] 21</code></pre>
<pre class="r"><code>print(n)</code></pre>
<pre><code>[1] 2068</code></pre>
<pre class="r"><code>print(p)</code></pre>
<pre><code>[1] 167178</code></pre>
<p>Lets now plot the distribution of factors for each drift event</p>
<pre class="r"><code># read factors
delta_df = as.data.frame(flash_fit$loadings$normalized.loadings[[2]])
colnames(delta_df)[1:K] = 1:K 

# gather the data.frame for plotting
delta_gath_df = delta_df %&gt;% 
                gather(K, value) %&gt;%
                filter(K!=1)

# plot the factors
K_ = K
p_fct = ggplot(delta_gath_df, aes(x=value)) + 
        scale_fill_manual(values = getPalette(K_)) +
        geom_histogram() + 
        facet_wrap(~factor(K, levels=2:K_), scales = &quot;free&quot;) + 
        labs(fill=&quot;K&quot;) + 
        scale_x_continuous(breaks = scales::pretty_breaks(n = 3)) +
        scale_y_continuous(breaks = scales::pretty_breaks(n = 3)) + 
        theme_bw()
p_fct</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-factors-1.png" width="768" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of flash-greedy-ld-viz-factors-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-factors-1.png" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/f5ef1af4d17d9469ed52811c88166186d0c3c993/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-factors-1.png" target="_blank">f5ef1af</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-15
</td>
</tr>
</tbody>
</table>
</details>
<p>We can see the later factors tend to get sparser but they still seem to contribute! Here is a plot of the “proportion of variance explained” of each factor:</p>
<pre class="r"><code>qplot(2:K, flash_fit$pve[2:K]) + ylab(&quot;Proportion of Varaince Explained&quot;) + xlab(&quot;K&quot;) + theme_bw()</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-pve-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of flash-greedy-ld-viz-pve-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-pve-1.png" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>print(flash_fit$pve)</code></pre>
<pre><code> [1] 0.4750002010 0.0221931227 0.0180913971 0.0103143286 0.0025255466
 [6] 0.0015774930 0.0045260494 0.0014121917 0.0012173854 0.0007676335
[11] 0.0006728921 0.0008111508 0.0002212648 0.0003887796 0.0002380212
[16] 0.0002352021 0.0001895606 0.0002282840 0.0002907459 0.0001156450
[21] 0.0001605612</code></pre>
<p>It looks like the PVE drops off at around 11 or so? I setup the <code>flashier</code> run so it estimates a SNP specific precision term. Here is a histogram of fitted variances:</p>
<pre class="r"><code>qplot(1/flash_fit$fit$est.tau) + xlab(&quot;Estimated Variance&quot;) + ylab(&quot;Count&quot;) + theme_bw()</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-var-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of flash-greedy-ld-viz-var-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-var-1.png" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
</tr>
</tbody>
</table>
</details>
<p>Lets now look the the fitted means:</p>
<pre class="r"><code>qplot(delta_df$`1`) + xlab(&quot;Estimated Mean&quot;) + ylab(&quot;Count&quot;) + theme_bw()</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-mean-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of flash-greedy-ld-viz-mean-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mean-1.png" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
</tr>
</tbody>
</table>
</details>
<p>The mean seems smaller than I would have expected but thats probably because its normalized to have unit norm?! These plots looks about reasonable as each of the SNP variances should roughly be interpreted as average heterozygosity <span class="math inline">\(\approx 2p(1-p)\)</span>? The mean term should roughly be interpreted as the mean minor allele frequency at the SNP and thus we should see a quadratic relationship with the estimated variance:</p>
<pre class="r"><code>d1 = flash_fit$loadings$scale.constant[1]
p_mv = qplot(sqrt(d1) * delta_df$`1`, 1/flash_fit$fit$est.tau, alpha=.3) + 
       xlab(&quot;Rescaled Estimated Mean&quot;) + ylab(&quot;Estimated Variance&quot;) + 
       scale_alpha(guide = &quot;none&quot;) + 
       stat_function(fun = function(x){return(2*x*(1-x))}, color=&quot;red&quot;) + 
       xlim(0, .5) + 
       theme_bw()
p_mv</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-mv-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of flash-greedy-ld-viz-mv-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/63e7173f9a33d2a087e6f16d90cd929eb422bb6b/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mv-1.png" target="_blank">63e7173</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-mv-1.png" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
</tr>
</tbody>
</table>
</details>
<p>I need to better understand the <code>flashier</code> output but it seems that when I scale the estimated mean by the square root of the output scale constant and plot this rescaled mean against the estimated variance, most of the SNPs have a mean-variance relationship expected under a simple Binomial model for the genotypes i.e. <span class="math inline">\(y_{ij} \sim Binomial(2, p_{ij})\)</span>. Though, there are indeed many SNPs that have a a much larger variance than expected under this simple “Hardy-Weinberg” model. I wonder if there is anything “special” going on with those SNPs (this is something to follow up on). Lets now take a look at the loadings. First we setup a data.frame that we can work with:</p>
<pre class="r"><code># read the meta data
meta_df = read.table(&quot;../data/meta/HumanOriginsPublic2068_maf_geno_ldprune.meta&quot;, sep=&quot; &quot;, header=T)

# setup loadings data.frame
l_df = as.data.frame(flash_fit$loadings$normalized.loadings[[1]])
K = ncol(l_df)
l_df$iid = as.vector(meta_df$iid) # individual ids
l_df$clst = meta_df$clst # population labels
pops = unique(l_df$clst) # all unique pop labels

# join with the meta data
l_df = l_df %&gt;% inner_join(meta_df, on=&quot;clst&quot;)
l_df = l_df %&gt;% arrange(region, clst) # sort by region then by population
l_df$iid = factor(l_df$iid, levels = l_df$iid) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

head(l_df)</code></pre>
<pre><code>           1           2            3          4            5          6
1 0.02198997 0.009454832 4.485180e-06 0.02211841 3.303098e-05 0.01764253
2 0.02198997 0.009511229 4.406670e-06 0.02247997 4.952488e-05 0.01608845
3 0.02198997 0.006147078 4.086449e-06 0.02482062 2.726321e-05 0.01692735
4 0.02198997 0.009103984 4.338176e-06 0.02293795 2.797966e-05 0.01742585
5 0.02198997 0.009529573 4.394827e-06 0.02267025 3.845100e-05 0.01450389
6 0.02198997 0.009243629 4.460459e-06 0.02215776 2.795106e-05 0.01760249
            7            8            9           10           11
1 0.004374536 1.089539e-03 2.812993e-05 3.470028e-05 3.866924e-05
2 0.003241962 4.231122e-04 2.932860e-05 3.615817e-05 3.888100e-05
3 0.004951296 9.893482e-05 2.583221e-05 3.555401e-05 3.744459e-05
4 0.004661258 1.784788e-04 2.702595e-05 3.411185e-05 3.742829e-05
5 0.004437185 8.043608e-04 2.643633e-05 3.423255e-05 3.784264e-05
6 0.005064440 8.324509e-05 2.791866e-05 3.363985e-05 3.856931e-05
          12           13           14         15           16
1 0.05017069 8.140998e-05 4.952727e-05 0.08068407 7.771326e-05
2 0.04589132 8.638569e-05 5.161609e-05 0.05897009 7.994484e-05
3 0.05074213 8.358567e-05 5.114096e-05 0.08024062 7.892103e-05
4 0.05519966 7.798145e-05 4.992192e-05 0.09129734 8.299529e-05
5 0.05242051 7.961690e-05 4.865967e-05 0.09571542 8.121683e-05
6 0.05152888 8.057720e-05 5.030271e-05 0.07831456 7.913664e-05
            17           18           19           20           21
1 7.842225e-05 1.019054e-04 2.147227e-04 8.793061e-05 8.502300e-05
2 8.227041e-05 8.412758e-05 1.665734e-04 8.801740e-05 1.154576e-04
3 9.672675e-05 8.794415e-05 1.557965e-04 8.469595e-05 2.321882e-04
4 8.006631e-05 1.754711e-04 9.898022e-05 8.627449e-05 3.734673e-04
5 7.616521e-05 8.802899e-05 1.127468e-04 8.540387e-05 6.950001e-03
6 8.043370e-05 8.734172e-05 2.415143e-04 8.651963e-05 8.910156e-05
            iid     clst region country  lat lon    clst2
1 Algerian43A22 Algerian Africa Algeria 36.8   3 Algerian
2 Algerian43A21 Algerian Africa Algeria 36.8   3 Algerian
3 Algerian43A34 Algerian Africa Algeria 36.8   3 Algerian
4 Algerian43A13 Algerian Africa Algeria 36.8   3 Algerian
5 Algerian43A24 Algerian Africa Algeria 36.8   3 Algerian
6 Algerian43A32 Algerian Africa Algeria 36.8   3 Algerian</code></pre>
<p>Its hard to find a color scale that can sufficiently visualize all of the loadings in a single plot. Instead I just split the loadings up into two plots (K=2,…,11) and (K=12,…,21). Lets first visualize loadings 2 through 12:</p>
<pre class="r"><code># gather the data.frame for plotting
l_gath_df = l_df %&gt;% 
            gather(K, value, -iid, -clst, -region, -country, -lat, -lon, -clst2) %&gt;% 
            filter(K %in% paste0(2:11))

# Africa
africa_pops = get_pops(meta_df, &quot;Africa&quot;)
p_africa = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;Africa&quot;), 
                                   africa_pops, colset=&quot;Set3&quot;, label_size=5) +
           ggtitle(&quot;Africa&quot;) + theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, &quot;America&quot;)
p_america = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;America&quot;), 
                                    america_pops, colset=&quot;Set3&quot;, label_size=5) + 
            ggtitle(&quot;America&quot;) + theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, &quot;CentralAsiaSiberia&quot;)
p_central_asia_siberia = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;CentralAsiaSiberia&quot;), 
                                                 central_asia_siberia_pops, colset=&quot;Set3&quot;, label_size=5) + 
                         ggtitle(&quot;CentralAsiaSiberia&quot;) + theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, &quot;EastAsia&quot;)
p_east_asia = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;EastAsia&quot;), 
                                      east_asia_pops, colset=&quot;Set3&quot;, label_size=5) + 
              ggtitle(&quot;EastAsia&quot;) + theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, &quot;SouthAsia&quot;)
p_south_asia= positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;SouthAsia&quot;),
                                      south_asia_pops, colset=&quot;Set3&quot;, label_size=5) + 
              ggtitle(&quot;SouthAsia&quot;) + theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, &quot;WestEurasia&quot;)
p_west_eurasia = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;WestEurasia&quot;), 
                                         west_eurasia_pops, colset=&quot;Set3&quot;, label_size=5) + 
                 ggtitle(&quot;WestEurasia&quot;) + theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, &quot;Oceania&quot;)
p_oceania = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;Oceania&quot;), 
                                    oceania_pops, colset=&quot;Set3&quot;, label_size=5) + 
            ggtitle(&quot;Oceania&quot;) + theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = &quot;v&quot;) 
p</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-2-11-1.png" width="816" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of flash-greedy-ld-viz-loadings-2-11-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-2-11-1.png" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
</tr>
</tbody>
</table>
</details>
<p>Lets now visualize loadings 12 to 21 (be careful: there is no connection to the colors in the last plot):</p>
<pre class="r"><code># gather the data.frame for plotting
l_gath_df = l_df %&gt;% 
            gather(K, value, -iid, -clst, -region, -country, -lat, -lon, -clst2) %&gt;% 
            filter(K %in% paste0(12:21))

# Africa
africa_pops = get_pops(meta_df, &quot;Africa&quot;)
p_africa = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;Africa&quot;), 
                                   africa_pops, colset=&quot;Set3&quot;, label_size=5) +
           ggtitle(&quot;Africa&quot;) + theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, &quot;America&quot;)
p_america = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;America&quot;), 
                                    america_pops, colset=&quot;Set3&quot;, label_size=5) + 
            ggtitle(&quot;America&quot;) + theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, &quot;CentralAsiaSiberia&quot;)
p_central_asia_siberia = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;CentralAsiaSiberia&quot;), 
                                                 central_asia_siberia_pops, colset=&quot;Set3&quot;, label_size=5) + 
                         ggtitle(&quot;CentralAsiaSiberia&quot;) + theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, &quot;EastAsia&quot;)
p_east_asia = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;EastAsia&quot;), 
                                      east_asia_pops, colset=&quot;Set3&quot;, label_size=5) + 
              ggtitle(&quot;EastAsia&quot;) + theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, &quot;SouthAsia&quot;)
p_south_asia= positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;SouthAsia&quot;),
                                      south_asia_pops, colset=&quot;Set3&quot;, label_size=5) + 
              ggtitle(&quot;SouthAsia&quot;) + theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, &quot;WestEurasia&quot;)
p_west_eurasia = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;WestEurasia&quot;), 
                                         west_eurasia_pops, colset=&quot;Set3&quot;, label_size=5) + 
                 ggtitle(&quot;WestEurasia&quot;) + theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, &quot;Oceania&quot;)
p_oceania = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;Oceania&quot;), 
                                    oceania_pops, colset=&quot;Set3&quot;, label_size=5) + 
            ggtitle(&quot;Oceania&quot;) + theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = &quot;v&quot;) 
p</code></pre>
<p><img src="figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-12-21-1.png" width="816" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of flash-greedy-ld-viz-loadings-12-21-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/flash-greedy-ld-viz-loadings-12-21-1.png" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
</tr>
</tbody>
</table>
</details>
<p>Its kinda interesting to see that some populations have zero loading on later factors. This would be difficult to visualize see if using a single plot for all the factors.</p>
</div>
<div id="admixture" class="section level3">
<h3>ADMIXTURE</h3>
<p>Lets visualize ADMIXTURE with 9 factors which should roughly align to the first plot i.e. <code>FLASH</code> with 2,…,11 (be careful: there is no connection to the colors in the last plot):</p>
<pre class="r"><code>l_df = read.table(&quot;../output/admixture/hoa_global_ld/HumanOriginsPublic2068_maf_geno_ldprune.K9r1.Q&quot;, sep=&quot; &quot;, header=F)
K = ncol(l_df)

l_df$iid = as.vector(meta_df$iid) # individual ids
l_df$clst = meta_df$clst # population labels

# join with the meta data
l_df = l_df %&gt;% inner_join(meta_df, on=&quot;clst&quot;)
l_df = l_df %&gt;% arrange(region, clst) # sort by region then by population
l_df$iid = factor(l_df$iid, levels = l_df$iid) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

# gather the data.frame for plotting
l_gath_df = l_df %&gt;% 
            gather(K, value, -iid, -clst, -region, -country, -lat, -lon, -clst2)

# Africa
africa_pops = get_pops(meta_df, &quot;Africa&quot;)
p_africa = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;Africa&quot;), 
                                   africa_pops, colset=&quot;Set3&quot;, label_size=5) +
           ggtitle(&quot;Africa&quot;) + theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, &quot;America&quot;)
p_america = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;America&quot;), 
                                    america_pops, colset=&quot;Set3&quot;, label_size=5) + 
            ggtitle(&quot;America&quot;) + theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, &quot;CentralAsiaSiberia&quot;)
p_central_asia_siberia = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;CentralAsiaSiberia&quot;), 
                                                 central_asia_siberia_pops, colset=&quot;Set3&quot;, label_size=5) + 
                         ggtitle(&quot;CentralAsiaSiberia&quot;) + theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, &quot;EastAsia&quot;)
p_east_asia = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;EastAsia&quot;), 
                                      east_asia_pops, colset=&quot;Set3&quot;, label_size=5) + 
              ggtitle(&quot;EastAsia&quot;) + theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, &quot;SouthAsia&quot;)
p_south_asia= positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;SouthAsia&quot;),
                                      south_asia_pops, colset=&quot;Set3&quot;, label_size=5) + 
              ggtitle(&quot;SouthAsia&quot;) + theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, &quot;WestEurasia&quot;)
p_west_eurasia = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;WestEurasia&quot;), 
                                         west_eurasia_pops, colset=&quot;Set3&quot;, label_size=5) + 
                 ggtitle(&quot;WestEurasia&quot;) + theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, &quot;Oceania&quot;)
p_oceania = positive_structure_plot(l_gath_df %&gt;% filter(region == &quot;Oceania&quot;), 
                                    oceania_pops, colset=&quot;Set3&quot;, label_size=5) + 
            ggtitle(&quot;Oceania&quot;) + theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = &quot;v&quot;) 
p</code></pre>
<p><img src="figure/hoa_global.Rmd/admixtrue-ld-viz-loadings-9-1.png" width="816" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of admixtrue-ld-viz-loadings-9-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jhmarcus/drift-workflow/blob/38b57c54997e44c495298de7423104f1749299a1/docs/figure/hoa_global.Rmd/admixtrue-ld-viz-loadings-9-1.png" target="_blank">38b57c5</a>
</td>
<td style="text-align:left;">
jhmarcus
</td>
<td style="text-align:left;">
2019-02-24
</td>
</tr>
</tbody>
</table>
</details>
<p>There is a lot that one can compare between the ADMIXTURE and FLASH results. A high level observation seems that the ADMIXTURE results look a bit more clustered i.e. the Americas and East Asia look like they are explained mostly by 1 or 2 factors whereas FLASH uses 3-4. Its hard to tell be it seems that this is true in many of the super regions … ADMIXTURE tends use fewer factors to explain population structure in each region, leading to a more clustered result?</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS  10.14.2

Matrix products: default
BLAS/LAPACK: /Users/jhmarcus/miniconda3/lib/R/lib/libRblas.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2.2     RColorBrewer_1.1-2 dplyr_0.7.6       
[4] tidyr_0.8.1        ggplot2_3.0.0     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0        compiler_3.5.1    pillar_1.3.0     
 [4] git2r_0.23.0      plyr_1.8.4        workflowr_1.1.1  
 [7] bindr_0.1.1       R.methodsS3_1.7.1 R.utils_2.7.0    
[10] tools_3.5.1       digest_0.6.18     evaluate_0.12    
[13] tibble_1.4.2      gtable_0.2.0      pkgconfig_2.0.1  
[16] rlang_0.3.1       yaml_2.2.0        xfun_0.4         
[19] flashier_0.1.0    withr_2.1.2       stringr_1.3.1    
[22] knitr_1.21        cowplot_0.9.4     rprojroot_1.3-2  
[25] grid_3.5.1        tidyselect_0.2.4  glue_1.3.0       
[28] R6_2.3.0          rmarkdown_1.11    reshape2_1.4.3   
[31] purrr_0.2.5       magrittr_1.5      whisker_0.3-2    
[34] backports_1.1.2   scales_0.5.0      htmltools_0.3.6  
[37] assertthat_0.2.0  colorspace_1.3-2  labeling_0.3     
[40] stringi_1.2.4     lazyeval_0.2.1    munsell_0.5.0    
[43] crayon_1.3.4      R.oo_1.22.0      </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
