---
title: "Human Origins Array Global Data"
author: "jhmarcus"
date: "2019-03-04"
output: workflowr::wflow_html
---

Here I explore basic properties of the Human Origins Array dataset. I downloaded the data from:

https://reich.hms.harvard.edu/sites/reich.hms.harvard.edu/files/inline-files/NearEastPublic.tar.gz

I subsequently converted the `eigenstrat` files to `plink` format using the following parameter file and `convertf` command:

```
genotypename:   HumanOriginsPublic2068.geno
snpname:    HumanOriginsPublic2068.snp
indivname:  HumanOriginsPublic2068.ind
outputformat:   PACKEDPED
genotypeoutname:    HumanOriginsPublic2068.bed
snpoutname: HumanOriginsPublic2068.bim
indivoutname:   HumanOriginsPublic2068.fam
familynames:    NO
```

```
convertf -p eig2plink.par
```

I then removed the sex chromosomes using the the following `plink` command:

```
plink --bfile HumanOriginsPublic2068 --make-bed --autosome --out HumanOriginsPublic2068_auto
```

# Imports

Lets import some needed packages:

```{r imports, warning=FALSE, message=FALSE}
library(ggplot2)
library(tidyr)
library(dplyr)
library(lfa)
```

# Read Genotypes

Here I read the full genotype matrix of the Human Origins dataset:

```{r genotypes, warning=FALSE, message=FALSE}
Y = t(lfa:::read.bed("../data/raw/NearEastPublic/HumanOriginsPublic2068_auto"))

# number of individuals
n = nrow(Y)

# number of SNPs
p = ncol(Y)
```



# Missingness per SNP

Here I compute the missingness per SNP:

```{r miss-per-snp, warning=FALSE, message=FALSE}
n_miss_snp = colSums(is.na(Y))
p_snpmss = qplot(n_miss_snp / n, bins=100) + theme_bw() + 
           scale_x_continuous(breaks = pretty(n_miss_snp / n, n = 10)) +
           xlab("Missingness Fraction") +
           ylab("Count")

p_snpmss
```

There are very few SNPs with high levels of missing data so we can use a very stringent missingness threshold without losing much information.

# Missingness per Individual

Here I compute the missingness per individual:

```{r miss-per-ind, warning=FALSE, message=FALSE}
n_miss_ind = rowSums(is.na(Y))
p_indmss = qplot(n_miss_ind / p) + theme_bw() +
           xlab("Missingness Fraction") +
           ylab("Count")
p_indmss
```

It seems like a few individuals are missing about 20000 of their SNPs which is a bit worrisome maybe they should be removed from the analysis? For now I will in include them and see if they pop up as any outliers in the PCs.

# Missingness per Population

Here I compute the missingness fraction per population:

```{r miss-per-pop, warning=FALSE, message=FALSE, fig.width=11, fig.height=7}
# meta data
meta_df = read.table("../data/meta/HumanOriginsPublic2068.meta", sep="\t", header=T)
meta_df$miss_frac = n_miss_ind / p

# average missingness per pop for sorting
avg_miss_df = meta_df %>% 
              group_by(Simple.Population.ID) %>% 
              summarise(avg_miss=mean(miss_frac)) %>%
              arrange(desc(avg_miss)) 

# distribution of missingness per pop
p_popmss = ggplot(meta_df, aes(x=factor(Simple.Population.ID, 
                                        levels=avg_miss_df$Simple.Population.ID), 
                               y=miss_frac)) + 
           geom_boxplot() +
           theme_classic() +
           theme(axis.text.x = element_text(angle = 90, hjust = 1, size=6)) +
           xlab("Population") +
           ylab("Missingness Fraction")
p_popmss
```

# Missingness per Contributer

Here I compute the average missingness fraction per contributor:

```{r miss-per-con, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=8.5}
# average missingness per contributer for sorting
avg_miss_df = meta_df %>% 
              group_by(Contributor) %>% 
              summarise(avg_miss=mean(miss_frac)) %>%
              arrange(desc(avg_miss)) 

# distribution of missingness per contributer
p_conmss = ggplot(meta_df, aes(x=factor(Contributor, 
                                        levels=avg_miss_df$Contributor), 
                               y=miss_frac)) + 
           geom_boxplot() +
           theme_classic() +
           theme(axis.text.x = element_text(angle = 90, hjust = 1, size=6)) +
           xlab("Contributer") +
           ylab("Missingness Fraction")
p_conmss
```

It seems like there is some variation in the amount of missingness per pop and contributor (there might be some confounding there) but the total amount of missingness is so low I think it can be ignored?

# Filter

Given the above results here are the `plink` commands I ran to filter the data:

```
plink --bfile HumanOriginsPublic2068 --geno .005 --maf .05 --make-bed --autosome --out HumanOriginsPublic2068_auto_maf05_geno005
```

These filtering steps take us from 616938 to 343758 SNPs ... which still likely contains a lot of information about population structure.