---
title: "Human Origins Array Global Results"
author: "jhmarcus"
date: "2019-02-15"
output:
  workflowr::wflow_html:
    code_folding: show
---

This is an analysis of the full Human Origins dataset which includes 2068 sampled from around the world. I filtered out rare variants with global minor allele frequency less than 5%, removed any variants with a missingness fraction greater than 0.5%, and removed any SNPs on the sex chromosomes, resulting in 343758 SNPs ... see [Human Origins Array Global Data](data_hoa_global.html) for details on the data pre-processing.

# Imports

Lets import some needed packages:

```{r imports, warning=FALSE, message=FALSE}
library(ggplot2)
library(tidyr)
library(dplyr)
library(RColorBrewer)
library(biomaRt)
library(knitr)
source("../code/viz.R")
source("../code/flash_prep.R")
```

# FLASH-Greedy

Here is the `snakemake` rule I used for running `flashier`:

```
    run:
        R("""
          # read the genotype matrix
          Y = t(lfa:::read.bed('{params.bed}'))
          # number of individuals
          n = nrow(Y)
          
          # run greedy flash
          flash_fit = flashier::flashier(Y,
                                         greedy.Kmax=20,
                                         ash.param=list(method='fdr'),
                                         prior.type=c('nonnegative', 'normal.mixture'),
                                         var.type=2,
                                         fixed.factors=flashier::ones.factor(1),
                                         output.lvl=4,
                                         verbose.lvl=2)
          # save the rds
          saveRDS(flash_fit, '{output.rds}')
```

Lets first read the greedy `flashier` fit

```{r flash-greedy}
# read the flash fit output by snakemake
flash_fit = readRDS("../output/flash_greedy/hoa_global/HumanOriginsPublic2068_auto_maf05_geno005_mind02.rds")

snp_df = prepare_snp_data(flash_fit, "../data/datasets/hoa_global/HumanOriginsPublic2068_auto_maf05_geno005_mind02.bim")

# drift event loadings lfsr
#l_lfsr_df = data.frame(flash_fit$lfsr[[1]])
#colnames(l_lfsr_df) = 1:21

# drift event lfsr
#delta_lfsr_df = data.frame(flash_fit$lfsr[[2]])
#colnames(delta_lfsr_df) = 1:21
#delta_lfsr_df$chrom = bim_df$chrom
#delta_lfsr_df$pos = bim_df$pos
#delta_lfsr_df$rsid = bim_df$rsid

# number of drift events
K = ncol(flash_fit$loadings$normalized.loadings[[1]]) 

# number of individuals
n = nrow(flash_fit$loadings$normalized.loadings[[1]])

# number of SNPs
p = nrow(flash_fit$loadings$normalized.loadings[[2]])
print(K)
print(n)
print(p)
```

# PVEs
 
Here is a plot of the proportion of variance (PVE) explained by each drift event:

```{r flash-greedy-ld-viz-pve, warning=FALSE, message=FALSE}
p_pve = qplot(2:K, flash_fit$pve[2:K]) + 
        ylab("Proportion of Varaince Explained") + 
        xlab("K") + 
        theme_bw()
p_pve
print(flash_fit$pve)
```

It looks like the PVE drops off at around 14 or so? 

# Fitted Mean and Variance

I setup the `flashier` run so it estimates a SNP specific precision term. Here is a histogram of fitted variances:

```{r flash-greedy-viz-var, warning=FALSE, message=FALSE}
p_var = qplot(1/flash_fit$fit$est.tau) + 
        xlab("Estimated Variance") + 
        ylab("Count") + 
        theme_bw()
p_var
```

Lets now look the the fitted means:

```{r flash-greedy-viz-mean, warning=FALSE, message=FALSE}
mu = sqrt(flash_fit$loadings$scale.constant[1]) * delta_df$`1`
p_mu = qplot(mu) + 
       xlab("Estimated Mean") + 
       ylab("Count") + 
       theme_bw()
p_mu
```

The mean term should roughly be interpreted as the minor allele frequency each SNP. I estimated the minor allele frequency using the raw genotypes an compared it to the fitted mean:

```{r mean-vs-maf, warning=FALSE, message=FALSE}
maf_df = read.table("../data/datasets/hoa_global/HumanOriginsPublic2068_auto_maf05_geno005.frq", 
                    stringsAsFactors=F, 
                    header=T)

maf_df$mu = mu

p_mu_maf = ggplot(maf_df %>% sample_n(10000), aes(x=MAF, y=mu)) + 
           geom_point() +
           xlab("MAF") + 
           ylab("Estimated Mean") + 
           theme_bw() +
           xlim(0.0, .5) +
           ylim(0.0, .5) +
           geom_abline(slope=1, intercept=0)

p_mu_maf
```

Its looks like the estimated mean is matching the MAF but is biased ... it tends to be underestimating the MAF. We should see a quadratic relationship with the estimated variance:

```{r flash-greedy-viz-mv, warning=FALSE, message=FALSE}
d1 = flash_fit$loadings$scale.constant[1]
mv_df = data.frame(var=1/flash_fit$fit$est.tau, mu=mu, chrom=bim_df$chrom)
p_mv = ggplot(mv_df, aes(x=mu, y=var)) + 
       geom_point() + 
       xlab("Estimated Mean") + ylab("Estimated Variance") + 
       scale_alpha(guide = "none") + 
       stat_function(fun = function(x){return(2*x*(1-x))}, color="red") + 
       xlim(0, .5) + 
       theme_bw()

p_mv
```

Most of the SNPs have a mean-variance relationship expected under a simple Binomial model for the genotypes i.e. $y_{ij} \sim Binomial(2, p_{ij})$. I wonder if there is anything "special" going on with the high variance SNP (will explore this later). I'm not sure why there is a sharp cuttof at ~.35.

# Drift Event Distributions

Lets now plot the distribution of drift events:

```{r flash-greedy-viz-factors, warning=FALSE, message=FALSE, fig.width=8, fig.height=6}
# gather the data.frame for plotting
delta_gath_df = delta_df %>% 
                gather(K, value, -chrom, -pos, -rsid) %>%
                filter(K!=1)

# plot the factors
K_ = K
p_de = ggplot(delta_gath_df, aes(x=value)) + 
       geom_histogram() + 
       facet_wrap(~factor(K, levels=2:K_), scales = "free") + 
       labs(fill="K") + 
       scale_x_continuous(breaks = scales::pretty_breaks(n = 3)) +
       scale_y_continuous(breaks = scales::pretty_breaks(n = 3)) + 
       theme_bw()

p_de
```

We can see the lower PVE drift events tend to get sparser!

# Drift Event Local False Sign Rates

Here histograms of the lfsrs for each drift event:

```{r flash-lfsr-factors-viz}
delta_lfsr_gath_df = delta_lfsr_df %>% 
                     gather(K, value, -chrom, -pos, -rsid) %>%
                     filter(K %in% paste0(2:21))

p_lfsr = ggplot(delta_lfsr_gath_df, aes(x=value)) + 
         geom_histogram() + 
         facet_wrap(~factor(K, levels=2:K_), scales = "free") + 
         labs(fill="K") + 
         scale_x_continuous(breaks = scales::pretty_breaks(n = 3)) +
         scale_y_continuous(breaks = scales::pretty_breaks(n = 3)) + 
         theme_bw() + 
         xlab("Local False Sign Rate")
p_lfsr
```

It is interesting to see the lower PVE factors (around 13 onwards) shift to having many SNPs with high uncertainty of the sign of the drift event.

# Drift Event Loadings (2-11)

Lets now take a look at the drift event loadings. First we setup the data:

```{r flash-greedy-data-loadings, warning=FALSE, message=FALSE}
# read the meta data
meta_df = read.table("../data/meta/HumanOriginsPublic2068.meta", sep="\t", header=T)

# setup loadings data.frame
l_df = as.data.frame(flash_fit$loadings$normalized.loadings[[1]])
inds = read.table("../data/datasets/hoa_global/HumanOriginsPublic2068_auto_maf05_geno005_mind02.fam", 
                  header=F, stringsAsFactors=F) %>% pull(V2)
l_df$ID = inds
l_df = l_df %>% inner_join(meta_df, by="ID")
K = ncol(l_df)
#l_df = cbind(l_df, meta_df)

# all unique pop labels
pops = unique(l_df$Simple.Population.ID) 

# join with the meta data
l_df = l_df %>% arrange(Region, Simple.Population.ID) # sort by region then by population
l_df$ID = factor(l_df$ID, levels = l_df$ID) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

head(l_df)
```

Its hard to find a color scale that can sufficiently visualize all of the loadings in a single plot. Instead I just split the loadings up into two plots (K=2,...,11) and (K=12,...,21). Lets first visualize loadings 2 through 12:

```{r flash-greedy-viz-loadings-2-11, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=12}
# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, 
                   -ID,
                   -Verbose.Population.ID, 
                   -Simple.Population.ID, 
                   -Region, -Country,
                   -Latitude,
                   -Longitude,
                   -Samples,
                   -Passed.QC,
                   -Contributor) %>% 
            filter(K %in% paste0(2:11))

# Africa
africa_pops = get_pops(meta_df, "Africa")
p_africa = positive_structure_plot(gath_df=l_gath_df %>% 
                                   filter(Region == "Africa"), 
                                   colset="Set3",
                                   facet_levels=africa_pops,
                                   facet_grp="Simple.Population.ID", 
                                   label_size=5) +
           ggtitle("Africa") + 
           theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, "America")
p_america = positive_structure_plot(gath_df=l_gath_df %>% 
                                    filter(Region == "America"), 
                                    colset="Set3",
                                    facet_levels=america_pops,
                                    facet_grp="Simple.Population.ID", 
                                    label_size=5) + 
            ggtitle("America") + 
            theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, "CentralAsiaSiberia")
p_central_asia_siberia = positive_structure_plot(gath_df=l_gath_df %>% 
                                                 filter(Region == "CentralAsiaSiberia"), 
                                                 colset="Set3",
                                                 facet_levels=central_asia_siberia_pops,  
                                                 facet_grp="Simple.Population.ID",
                                                 label_size=5) + 
                         ggtitle("CentralAsiaSiberia") + 
                         theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, "EastAsia")
p_east_asia = positive_structure_plot(gath_df=l_gath_df %>% 
                                      filter(Region == "EastAsia"), 
                                      colset="Set3",
                                      facet_levels=east_asia_pops,  
                                      facet_grp="Simple.Population.ID",
                                      label_size=5) + 
              ggtitle("EastAsia") + 
              theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, "SouthAsia")
p_south_asia= positive_structure_plot(gath_df=l_gath_df %>% 
                                      filter(Region == "SouthAsia"),
                                      colset="Set3",
                                      facet_levels=south_asia_pops, 
                                      facet_grp="Simple.Population.ID",
                                      label_size=5) + 
              ggtitle("SouthAsia") + 
              theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, "WestEurasia")
p_west_eurasia = positive_structure_plot(gath_df=l_gath_df %>% 
                                         filter(Region == "WestEurasia"), 
                                         colset="Set3",
                                         facet_levels=west_eurasia_pops, 
                                         facet_grp="Simple.Population.ID",
                                         label_size=5) + 
                 ggtitle("WestEurasia") + 
                 theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, "Oceania")
p_oceania = positive_structure_plot(gath_df=l_gath_df %>% 
                                    filter(Region == "Oceania"), 
                                    colset="Set3",
                                    facet_levels=oceania_pops, 
                                    facet_grp="Simple.Population.ID",
                                    label_size=5) + 
            ggtitle("Oceania") + 
            theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia,
                       p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = "v") 
p
```

# Drift Event Loadings (12-21)

Lets now visualize loadings 12 to 21 (be careful: there is no connection to the colors in the last plot):

```{r flash-greedy-viz-loadings-12-21, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=12}
# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, 
                   -ID,
                   -Verbose.Population.ID, 
                   -Simple.Population.ID, 
                   -Region, -Country,
                   -Latitude,
                   -Longitude,
                   -Samples,
                   -Passed.QC,
                   -Contributor) %>% 
            filter(K %in% paste0(12:21))

# Africa
africa_pops = get_pops(meta_df, "Africa")
p_africa = positive_structure_plot(gath_df=l_gath_df %>% 
                                   filter(Region == "Africa"), 
                                   colset="Set3",
                                   facet_levels=africa_pops,
                                   facet_grp="Simple.Population.ID", 
                                   label_size=5) +
           ggtitle("Africa") + 
           theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, "America")
p_america = positive_structure_plot(gath_df=l_gath_df %>% 
                                    filter(Region == "America"), 
                                    colset="Set3",
                                    facet_levels=america_pops,
                                    facet_grp="Simple.Population.ID", 
                                    label_size=5) + 
            ggtitle("America") + 
            theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, "CentralAsiaSiberia")
p_central_asia_siberia = positive_structure_plot(gath_df=l_gath_df %>% 
                                                 filter(Region == "CentralAsiaSiberia"), 
                                                 colset="Set3",
                                                 facet_levels=central_asia_siberia_pops,  
                                                 facet_grp="Simple.Population.ID",
                                                 label_size=5) + 
                         ggtitle("CentralAsiaSiberia") + 
                         theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, "EastAsia")
p_east_asia = positive_structure_plot(gath_df=l_gath_df %>% 
                                      filter(Region == "EastAsia"), 
                                      colset="Set3",
                                      facet_levels=east_asia_pops,  
                                      facet_grp="Simple.Population.ID",
                                      label_size=5) + 
              ggtitle("EastAsia") + 
              theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, "SouthAsia")
p_south_asia= positive_structure_plot(gath_df=l_gath_df %>% 
                                      filter(Region == "SouthAsia"),
                                      colset="Set3",
                                      facet_levels=south_asia_pops, 
                                      facet_grp="Simple.Population.ID",
                                      label_size=5) + 
              ggtitle("SouthAsia") + 
              theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, "WestEurasia")
p_west_eurasia = positive_structure_plot(gath_df=l_gath_df %>% 
                                         filter(Region == "WestEurasia"), 
                                         colset="Set3",
                                         facet_levels=west_eurasia_pops, 
                                         facet_grp="Simple.Population.ID",
                                         label_size=5) + 
                 ggtitle("WestEurasia") + 
                 theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, "Oceania")
p_oceania = positive_structure_plot(gath_df=l_gath_df %>% 
                                    filter(Region == "Oceania"), 
                                    colset="Set3",
                                    facet_levels=oceania_pops, 
                                    facet_grp="Simple.Population.ID",
                                    label_size=5) + 
            ggtitle("Oceania") + 
            theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia,
                       p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = "v") 
p
```

Its kinda interesting to see that some populations have zero loading on later factors. Its also interesting to see a lot of population specific factors arising. This would be difficult to visualize see if using a single plot for all the factors. There is a light purple loading that seems to appear in many populations that don't obviously seem connected. I need to follow up on this.

# Drift Event Loadings Local False Sign Rates

```{r flash-lfsr-loadings-viz}
l_lfsr_gath_df = l_lfsr_df %>% 
                 gather(K, value) %>%
                 filter(K %in% paste0(2:21))

p_lfsr = ggplot(l_lfsr_gath_df, aes(x=value)) + 
         geom_histogram() + 
         facet_wrap(~factor(K, levels=2:K_), scales = "free") + 
         labs(fill="K") + 
         scale_x_continuous(breaks = scales::pretty_breaks(n = 3)) +
         scale_y_continuous(breaks = scales::pretty_breaks(n = 3)) + 
         theme_bw() + 
         xlab("Local False Sign Rate")
p_lfsr
```

Its very interesting to see that the lfsr seem very bi-model for the loadings for many of the drift events.

# LFSR Tree

Here I visualize the processed lfsr data that would go into constructing a tree from the loadings. I construct a binary matrix whose elements are 1 if the the fraction of individuals with lfsr < .05 for a given drift event is > 20% and 0 otherwise. In otherwords the matrix gives a sense if the drift event is present / absent in each population. I organize this matrix so that its sorted in a similar way to the above bar charts:

```{r flash-lfsr-tree-viz, warning=FALSE, message=FALSE, fig.width=12, fig.height=10}
# add meta data to lsfr
l_lfsr_df = cbind(l_lfsr_df, meta_df)

# gather the data.frame for plotting
l_lfsr_gath_df = l_lfsr_df %>% 
                 gather(K, value, 
                        -ID,
                        -Verbose.Population.ID, 
                        -Simple.Population.ID, 
                        -Region, -Country,
                        -Latitude,
                        -Longitude,
                        -Samples,
                        -Passed.QC,
                        -Contributor) %>%
                    filter(K!="1")

# construct binary indicator df 
lfsr_thresh = .05
lfsr_pop_frac = .2
b_df = l_lfsr_gath_df %>% 
       group_by(Simple.Population.ID, K) %>% 
       summarise(I=ifelse((sum(value <= lfsr_thresh) / n()) <= lfsr_pop_frac, 0, 1),
                 Region = unique(Region), 
                 Longitude=unique(Longitude), 
                 Latitude=unique(Latitude)) 

# plot the raw indicator matrix organized by region
p = ggplot(b_df, aes(x=factor(K, paste0(2:21)), 
                     y=reorder(Simple.Population.ID,
                             desc(Latitude)), fill=factor(I))) + 
    scale_fill_manual(values=c("white", "black")) + 
    geom_tile() + 
    facet_wrap(Region~., scales="free") + 
    theme_bw() +
    theme(axis.text.y=element_text(size=5), 
          axis.text.x=element_text(size=8)) + 
    xlab("K") +
    ylab("Population") +
    labs(fill="LFSR Indicator")
p
```

# PCA

Here are proportions of variances explained by each PC:

```{r pca-viz-pves, warning=FALSE, message=FALSE}
pca_fit = readRDS("../output/pca/hoa_global/HumanOriginsPublic2068_auto_maf05_geno005.rds")
lamb2 = pca_fit$d^2

p_pve = qplot(x=1:length(lamb2), y=lamb2 / sum(lamb2)) + 
        xlab("PC") +
        ylab("PVE") +
        theme_bw()
        
p_pve
```

Lets plot the PC loadings

```{r pca-viz-loadings, warning=FALSE, message=FALSE, fig.width=10, fig.height=7}
l_df = cbind(data.frame(pca_fit$v), meta_df)
colnames(l_df)[1:20] = 1:20
l_df = l_df %>% rowwise() %>% mutate(lab=substr(Simple.Population.ID, 1, 3))

# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, 
                   -ID, 
                   -Verbose.Population.ID, 
                   -Simple.Population.ID, 
                   -Region,
                   -Country, 
                   -Latitude, 
                   -Longitude, 
                   -Samples,
                   -Passed.QC,  
                   -Contributor, 
                   -lab) 

pc_df = l_gath_df %>% 
        inner_join(l_gath_df, by="ID") %>% 
        filter((K.x == "1" & K.y == "2") | 
               (K.x == "3" & K.y == "4") |
               (K.x == "5" & K.y == "6") |
               (K.x == "7" & K.y == "8") |
               (K.x == "9" & K.y == "10") |
               (K.x == "11" & K.y == "12") |
               (K.x == "13" & K.y == "14") |
               (K.x == "15" & K.y == "16") |
               (K.x == "17" & K.y == "18") |
               (K.x == "19" & K.y == "20")) %>%
        mutate(Region=Region.x)

p_pca = ggplot(pc_df, aes(x=value.x, y=value.y, label=lab.x, color=Region)) +
        geom_text(size=2) +
        scale_color_brewer(palette = "Set1") + 
        facet_wrap(factor(K.x, levels=seq(1,20,2))~factor(K.y, levels=seq(2,20,2)), scales="free") + 
        theme_bw() + 
        xlab("PCx") +
        ylab("PCy")
        
p_pca
```

The classic horseshoe pattern is obtained that we often see for PCA on global populations ... then different regions and populations have high loadings in subsequent PCs i.e. PC3~Ocenica, PC4~Americas, etc. I now see the advantage of the bar plot for visualizing lower PVE factors. Its "OK" if we only focus on PC1 / PC2 but the bar plot has a massive advantage when trying to compare many factors.

# ADMIXTURE

Lets visualize ADMIXTURE with 9 factors which should roughly align to the first plot i.e. `FLASH` with 2,...,11 (be careful: there is no connection to the colors in the last plot):

```{r admixture-viz-loadings-9, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=12}
l_df = read.table("../output/admixture/hoa_global/HumanOriginsPublic2068_auto_maf05_geno005_mind02.K9r1.Q", sep=" ", header=F)
K = ncol(l_df)
inds = read.table("../data/datasets/hoa_global/HumanOriginsPublic2068_auto_maf05_geno005_mind02.fam", 
                  header=F, stringsAsFactors=F) %>% pull(V2)
l_df$ID = inds
l_df = l_df %>% inner_join(meta_df, by="ID")
pops = unique(l_df$Simple.Population.ID) # all unique pop labels
l_df = l_df %>% arrange(Region, Simple.Population.ID) # sort by region then by population
l_df$ID = factor(l_df$ID, levels = l_df$ID) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, -ID, -Verbose.Population.ID, -Simple.Population.ID, 
                   -Region, -Country, -Latitude, -Longitude, -Samples,
                   -Passed.QC,  -Contributor) 

# Africa
africa_pops = get_pops(meta_df, "Africa")
p_africa = structure_plot(gath_df=l_gath_df %>% 
                                   filter(Region == "Africa"), 
                                   colset="Set3",
                                   facet_levels=africa_pops,
                                   facet_grp="Simple.Population.ID", 
                                   label_size=5, 
                                   fact_type="structure") +
           ggtitle("Africa") + 
           theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, "America")
p_america = structure_plot(gath_df=l_gath_df %>% 
                                    filter(Region == "America"), 
                                    colset="Set3",
                                    facet_levels=america_pops,
                                    facet_grp="Simple.Population.ID", 
                                    label_size=5,
                                    fact_type="structure") + 
            ggtitle("America") + 
            theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, "CentralAsiaSiberia")
p_central_asia_siberia = structure_plot(gath_df=l_gath_df %>% 
                                                 filter(Region == "CentralAsiaSiberia"), 
                                                 colset="Set3",
                                                 facet_levels=central_asia_siberia_pops,  
                                                 facet_grp="Simple.Population.ID",
                                                 label_size=5,
                                                 fact_type="structure") + 
                         ggtitle("CentralAsiaSiberia") + 
                         theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, "EastAsia")
p_east_asia = structure_plot(gath_df=l_gath_df %>% 
                                      filter(Region == "EastAsia"), 
                                      colset="Set3",
                                      facet_levels=east_asia_pops,  
                                      facet_grp="Simple.Population.ID",
                                      label_size=5,
                                      fact_type="structure") + 
              ggtitle("EastAsia") + 
              theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, "SouthAsia")
p_south_asia= structure_plot(gath_df=l_gath_df %>% 
                                      filter(Region == "SouthAsia"),
                                      colset="Set3",
                                      facet_levels=south_asia_pops, 
                                      facet_grp="Simple.Population.ID",
                                      label_size=5,
                                      fact_type="structure") + 
              ggtitle("SouthAsia") + 
              theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, "WestEurasia")
p_west_eurasia = structure_plot(gath_df=l_gath_df %>% 
                                         filter(Region == "WestEurasia"), 
                                         colset="Set3",
                                         facet_levels=west_eurasia_pops, 
                                         facet_grp="Simple.Population.ID",
                                         label_size=5,
                                         fact_type="structure") + 
                 ggtitle("WestEurasia") + 
                 theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, "Oceania")
p_oceania = structure_plot(gath_df=l_gath_df %>% 
                                    filter(Region == "Oceania"), 
                                    colset="Set3",
                                    facet_levels=oceania_pops, 
                                    facet_grp="Simple.Population.ID",
                                    label_size=5,
                                    fact_type="structure") + 
            ggtitle("Oceania") + 
            theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia,
                       p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = "v") 
p
```

There is a lot that one can compare between the ADMIXTURE and FLASH results. A high level observation seems that the ADMIXTURE results look a bit more clustered i.e. the Americas and East Asia look like they are explained mostly by 1 or 2 factors whereas FLASH uses 3-4. Its hard to tell be it seems that this is true in many of the super regions ...  ADMIXTURE tends use fewer factors to explain population structure in each region, leading to a more clustered result? 

# Outlier SNPs

Lets take a closer look at the drift factors to see if they are clustering in particular regions of the genome. As a first pass I take the top 5% of SNPs weighted on each drift event (to be clear I ignore the sign of each SNP). I would like to use the lfsr here but I will return to later. I then made a Manhatten plot for each chromosome and factor:

```{r flash-factors-chrom-viz, fig.width=8.5, fig.height=8}
manh_df = delta_lfsr_df %>% 
          group_by(chrom) %>% 
          summarise(chr_len=max(pos)) %>% 
          mutate(tot=cumsum(chr_len)-chr_len) %>%
          dplyr::select(-chr_len) %>%
          left_join(delta_lfsr_df, ., by=c("chrom"="chrom")) %>%
          arrange(chrom, pos) %>%
          mutate(BPcum=pos+tot)

manh_axis_df = manh_df %>% 
               group_by(chrom) %>% 
               summarize(center=(max(BPcum) + min(BPcum)) / 2)

manh_gath_df = manh_df %>% gather(K, value, -chrom, -pos, -rsid, -tot, -BPcum) %>% 
               filter(K %in% paste0(2:11)) %>% 
               filter(value < 1e-10) %>%
               filter(chrom %in% 1:22)
          
p = ggplot(manh_gath_df, aes(x=BPcum, y=-log10(value))) +
    geom_point(aes(color=as.factor(chrom)), alpha=.7, size=.5) +
    scale_color_manual(values=rep(c("grey", "orange"), 22)) +
    scale_x_continuous(label=manh_axis_df$chrom, breaks=manh_axis_df$center) +
    scale_y_continuous(expand=c(0, 0)) +     
    theme_bw() +
    theme(axis.title.x=element_blank(), axis.ticks.x=element_blank(),
          panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
    facet_grid(factor(K, levels=2:11)~., scales = "free") + 
    ylab("-log10(lfsr)") + 
    guides(alpha=F, color=F)
p
```

We can see there are some regions on the chromosomes that are peaky as well as some regions that have no "outliers" at all. I then took the top 5 outliers in each factor and annotated them with some functional information:

```{r tophits}
# SNP bioMart database
grch37_snp = useMart(biomart="ENSEMBL_MART_SNP", 
                     host="grch37.ensembl.org", 
                     path="/biomart/martservice",
                     dataset="hsapiens_snp")

# GENE bioMart database
grch37 = useMart(biomart="ENSEMBL_MART_ENSEMBL", 
                 host="grch37.ensembl.org", 
                 path="/biomart/martservice", 
                 dataset="hsapiens_gene_ensembl")

# top rank SNPs per each drift event
delta_tophit_df = delta_df %>%
                  gather(K, value, -chrom, -pos, -rsid) %>%
                  filter(K %in% paste0(2:21)) %>% 
                  group_by(K) %>%
                  top_n(5, value)

# SNP meta data
table1 = getBM(attributes=c("refsnp_id", 
                             "chrom_start", 
                             "minor_allele_freq",
                             "ensembl_gene_stable_id",
                             "consequence_type_tv", 
                             "associated_gene"), 
                filters = "snp_filter", 
                values = delta_tophit_df$rsid, 
                mart = grch37_snp)
table1$ensembl_gene_id = table1$ensembl_gene_stable_id

# GENE meta data
table2 = getBM(attributes = c("ensembl_gene_id", "external_gene_name", "description"),
               filters = "ensembl_gene_id", 
               values =  table1$ensembl_gene_stable_id, 
               mart = grch37)

# annotation data
anno_df = table1 %>% left_join(table2, on="ensembl_gene_id") %>% 
          mutate(rsid=refsnp_id) %>% 
          inner_join(delta_tophit_df, on="rsid") 

# unique genes
print(unique(anno_df$external_gene_name))

# formatted table
d = anno_df %>% 
    distinct(external_gene_name, .keep_all = T) %>% 
    dplyr::select(external_gene_name, K, rsid, consequence_type_tv) %>% 
    arrange(consequence_type_tv)
kable(d)
```

We can see many of these top outliers are in genes (but we have to consider the array design to know if this is unusual). Its cool to see a couple very interesting genes pop up including HERC2 (eye color) and SLC45A2 (skin color), EDAR (hair), ADH4 (alcohol dehydrogenase), ADH1B (alcohol dehydrogenase), which have been previously studied for their selection signatures. Its also fun to look at some of the `rsids` on `https://popgen.uchicago.edu/ggv/` to get a sense of what kind of allele frequency distributions define each factor.
