---
title: "Human Origins Array Global Results"
author: "jhmarcus"
date: "2019-02-15"
output:
  workflowr::wflow_html:
    code_folding: show
---

## Imports

Lets import some needed packages:

```{r imports, warning=FALSE, message=FALSE}
library(ggplot2)
library(tidyr)
library(dplyr)
library(RColorBrewer)
library(biomaRt)
library(knitr)
source("../code/viz.R")
```

## Human Origins Global (LD Pruned)

This an analysis of the full Human Origins dataset which includes 2068 sampled from around the world. I filtered out rare variants with global minor allele frequency less than 5%, removed any variants with a missingness level greater than 1%, and removed any SNPs on the sex chromosomes. I then LD pruned the SNPs using standard parameters in `plink`, resulting in 165468 SNPs.

### Greedy

Here is the `snakemake` rule I used for running `flashier`:

```
    run:
        R("""
          # read the genotype matrix
          Y = t(lfa:::read.bed('{params.bed}'))
          # number of individuals
          n = nrow(Y)
          
          # run greedy flash
          flash_fit = flashier::flashier(Y,
                                         greedy.Kmax=20,
                                         prior.type=c('nonnegative', 'point.normal'),
                                         var.type=2,
                                         fix.dim=list(1),
                                         fix.idx=list(1:n),
                                         fix.vals=list(rep(1, n)))
          # save the rds
          saveRDS(flash_fit, '{output.rds}')
          """)
```

Lets first read the greedy `flashier` fit

```{r flash-greedy-ld}
flash_fit = readRDS("../output/flash_greedy/hoa_global_ld/HumanOriginsPublic2068_maf_geno_auto_ldprune.rds")
K = ncol(flash_fit$loadings$normalized.loadings[[1]]) 
n = nrow(flash_fit$loadings$normalized.loadings[[1]])
p = nrow(flash_fit$loadings$normalized.loadings[[2]])
print(K)
print(n)
print(p)
```

Lets now plot the distribution of factors for each drift event

```{r flash-greedy-ld-viz-factors, warning=FALSE, message=FALSE, fig.width=8, fig.height=6}
# read factors
delta_df = as.data.frame(flash_fit$loadings$normalized.loadings[[2]])
colnames(delta_df)[1:K] = 1:K 

# add snp meta data
bim_df = read.table("../data/raw/NearEastPublic/HumanOriginsPublic2068_maf_geno_auto_ldprune.bim", header=F)
colnames(bim_df) = c("chrom", "rsid", "cm", "pos", "a1", "a2")
delta_df$chrom = bim_df$chrom
delta_df$pos = bim_df$pos
delta_df$rsid = bim_df$rsid

# gather the data.frame for plotting
delta_gath_df = delta_df %>% 
                gather(K, value, -chrom, -pos, -rsid) %>%
                filter(K!=1)

# plot the factors
K_ = K
p_fct = ggplot(delta_gath_df, aes(x=value)) + 
        geom_histogram() + 
        facet_wrap(~factor(K, levels=2:K_), scales = "free") + 
        labs(fill="K") + 
        scale_x_continuous(breaks = scales::pretty_breaks(n = 3)) +
        scale_y_continuous(breaks = scales::pretty_breaks(n = 3)) + 
        theme_bw()
p_fct
```

We can see the later factors tend to get sparser! Here is a plot of the "proportion of variance explained" of each factor:

```{r flash-greedy-ld-viz-pve, warning=FALSE, message=FALSE}
qplot(2:K, flash_fit$pve[2:K]) + ylab("Proportion of Varaince Explained") + xlab("K") + theme_bw()
print(flash_fit$pve)
```

It looks like the PVE drops off at around 14 or so? I setup the `flashier` run so it estimates a SNP specific precision term. Here is a histogram of fitted variances:

```{r flash-greedy-ld-viz-var, warning=FALSE, message=FALSE}
qplot(1/flash_fit$fit$est.tau) + xlab("Estimated Variance") + ylab("Count") + theme_bw()
```

Lets now look the the fitted means:

```{r flash-greedy-ld-viz-mean, warning=FALSE, message=FALSE}
mu = sqrt(flash_fit$loadings$scale.constant[1]) * delta_df$`1`
qplot(mu) + xlab("Estimated Mean") + ylab("Count") + theme_bw()
```

These plots looks about reasonable as each of the SNP variances should roughly be interpreted as average heterozygosity $\approx 2p(1-p)$? The mean term should roughly be interpreted as the mean minor allele frequency at the SNP and thus we should see a quadratic relationship with the estimated variance:

```{r flash-greedy-ld-viz-mv, warning=FALSE, message=FALSE}
d1 = flash_fit$loadings$scale.constant[1]
mv_df = data.frame(var=1/flash_fit$fit$est.tau, mu=mu, chrom=bim_df$chrom)
p_mv = ggplot(mv_df, aes(x=mu, y=var)) + 
       geom_point() + 
       xlab("Rescaled Estimated Mean") + ylab("Estimated Variance") + 
       scale_alpha(guide = "none") + 
       stat_function(fun = function(x){return(2*x*(1-x))}, color="red") + 
       xlim(0, .5) + 
       theme_bw()
p_mv
```

Most of the SNPs have a mean-variance relationship expected under a simple Binomial model for the genotypes i.e. $y_{ij} \sim Binomial(2, p_{ij})$. I wonder if there is anything "special" going on with the high variance SNP (will explore this later). Lets now take a look at the loadings. First we setup a data.frame that we can work with:

```{r flash-greedy-ld-data-loadings, warning=FALSE, message=FALSE}
# read the meta data
meta_df = read.table("../data/meta/HumanOriginsPublic2068.meta", sep="\t", header=T)

# setup loadings data.frame
l_df = as.data.frame(flash_fit$loadings$normalized.loadings[[1]])
K = ncol(l_df)
l_df = cbind(l_df, meta_df)
pops = unique(l_df$Simple.Population.ID) # all unique pop labels

# join with the meta data
l_df = l_df %>% arrange(Region, Simple.Population.ID) # sort by region then by population
l_df$ID = factor(l_df$ID, levels = l_df$ID) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

head(l_df)
```

Its hard to find a color scale that can sufficiently visualize all of the loadings in a single plot. Instead I just split the loadings up into two plots (K=2,...,11) and (K=12,...,21). Lets first visualize loadings 2 through 12:

```{r flash-greedy-ld-viz-loadings-2-11, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=12}
# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, -ID, -Verbose.Population.ID, -Simple.Population.ID, 
                   -Region, -Country, -Latitude, -Longitude, -Samples,
                   -Passed.QC,  -Contributor) %>% 
            filter(K %in% paste0(2:11))

# Africa
africa_pops = get_pops(meta_df, "Africa")
p_africa = positive_structure_plot(l_gath_df %>% filter(Region == "Africa"), 
                                   africa_pops, colset="Set3", label_size=5) +
           ggtitle("Africa") + theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, "America")
p_america = positive_structure_plot(l_gath_df %>% filter(Region == "America"), 
                                    america_pops, colset="Set3", label_size=5) + 
            ggtitle("America") + theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, "CentralAsiaSiberia")
p_central_asia_siberia = positive_structure_plot(l_gath_df %>% filter(Region == "CentralAsiaSiberia"), 
                                                 central_asia_siberia_pops, colset="Set3", label_size=5) + 
                         ggtitle("CentralAsiaSiberia") + theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, "EastAsia")
p_east_asia = positive_structure_plot(l_gath_df %>% filter(Region == "EastAsia"), 
                                      east_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("EastAsia") + theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, "SouthAsia")
p_south_asia= positive_structure_plot(l_gath_df %>% filter(Region == "SouthAsia"),
                                      south_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("SouthAsia") + theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, "WestEurasia")
p_west_eurasia = positive_structure_plot(l_gath_df %>% filter(Region == "WestEurasia"), 
                                         west_eurasia_pops, colset="Set3", label_size=5) + 
                 ggtitle("WestEurasia") + theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, "Oceania")
p_oceania = positive_structure_plot(l_gath_df %>% filter(Region == "Oceania"), 
                                    oceania_pops, colset="Set3", label_size=5) + 
            ggtitle("Oceania") + theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = "v") 
p
```

Lets now visualize loadings 12 to 21 (be careful: there is no connection to the colors in the last plot):

```{r flash-greedy-ld-viz-loadings-12-21, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=12}
# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, -ID, -Verbose.Population.ID, -Simple.Population.ID, 
                   -Region, -Country, -Latitude, -Longitude, -Samples,
                   -Passed.QC,  -Contributor) %>% 
            filter(K %in% paste0(12:21))

# Africa
africa_pops = get_pops(meta_df, "Africa")
p_africa = positive_structure_plot(l_gath_df %>% filter(Region == "Africa"), 
                                   africa_pops, colset="Set3", label_size=5) +
           ggtitle("Africa") + theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, "America")
p_america = positive_structure_plot(l_gath_df %>% filter(Region == "America"), 
                                    america_pops, colset="Set3", label_size=5) + 
            ggtitle("America") + theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, "CentralAsiaSiberia")
p_central_asia_siberia = positive_structure_plot(l_gath_df %>% filter(Region == "CentralAsiaSiberia"), 
                                                 central_asia_siberia_pops, colset="Set3", label_size=5) + 
                         ggtitle("CentralAsiaSiberia") + theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, "EastAsia")
p_east_asia = positive_structure_plot(l_gath_df %>% filter(Region == "EastAsia"), 
                                      east_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("EastAsia") + theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, "SouthAsia")
p_south_asia= positive_structure_plot(l_gath_df %>% filter(Region == "SouthAsia"),
                                      south_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("SouthAsia") + theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, "WestEurasia")
p_west_eurasia = positive_structure_plot(l_gath_df %>% filter(Region == "WestEurasia"), 
                                         west_eurasia_pops, colset="Set3", label_size=5) + 
                 ggtitle("WestEurasia") + theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, "Oceania")
p_oceania = positive_structure_plot(l_gath_df %>% filter(Region == "Oceania"), 
                                    oceania_pops, colset="Set3", label_size=5) + 
            ggtitle("Oceania") + theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = "v") 
p
```

Its kinda interesting to see that some populations have zero loading on later factors. Its also interesting to see a lot of population specific factors arising. This would be difficult to visualize see if using a single plot for all the factors. 

### ADMIXTURE

Lets visualize ADMIXTURE with 9 factors which should roughly align to the first plot i.e. `FLASH` with 2,...,11 (be careful: there is no connection to the colors in the last plot):

```{r admixture-ld-viz-loadings-9, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=12}
l_df = read.table("../output/admixture/hoa_global_ld/HumanOriginsPublic2068_maf_geno_auto_ldprune.K9r1.Q", sep=" ", header=F)
K = ncol(l_df)
l_df = cbind(l_df, meta_df)
pops = unique(l_df$Simple.Population.ID) # all unique pop labels
l_df = l_df %>% arrange(Region, Simple.Population.ID) # sort by region then by population
l_df$ID = factor(l_df$ID, levels = l_df$ID) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, -ID, -Verbose.Population.ID, -Simple.Population.ID, 
                   -Region, -Country, -Latitude, -Longitude, -Samples,
                   -Passed.QC,  -Contributor) 

# Africa
africa_pops = get_pops(meta_df, "Africa")
p_africa = positive_structure_plot(l_gath_df %>% filter(Region == "Africa"), 
                                   africa_pops, colset="Set3", label_size=5) +
           ggtitle("Africa") + theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, "America")
p_america = positive_structure_plot(l_gath_df %>% filter(Region == "America"), 
                                    america_pops, colset="Set3", label_size=5) + 
            ggtitle("America") + theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, "CentralAsiaSiberia")
p_central_asia_siberia = positive_structure_plot(l_gath_df %>% filter(Region == "CentralAsiaSiberia"), 
                                                 central_asia_siberia_pops, colset="Set3", label_size=5) + 
                         ggtitle("CentralAsiaSiberia") + theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, "EastAsia")
p_east_asia = positive_structure_plot(l_gath_df %>% filter(Region == "EastAsia"), 
                                      east_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("EastAsia") + theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, "SouthAsia")
p_south_asia= positive_structure_plot(l_gath_df %>% filter(Region == "SouthAsia"),
                                      south_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("SouthAsia") + theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, "WestEurasia")
p_west_eurasia = positive_structure_plot(l_gath_df %>% filter(Region == "WestEurasia"), 
                                         west_eurasia_pops, colset="Set3", label_size=5) + 
                 ggtitle("WestEurasia") + theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, "Oceania")
p_oceania = positive_structure_plot(l_gath_df %>% filter(Region == "Oceania"), 
                                    oceania_pops, colset="Set3", label_size=5) + 
            ggtitle("Oceania") + theme(plot.title = element_text(size=6))
# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = "v") 
p
```

There is a lot that one can compare between the ADMIXTURE and FLASH results. A high level observation seems that the ADMIXTURE results look a bit more clustered i.e. the Americas and East Asia look like they are explained mostly by 1 or 2 factors whereas FLASH uses 3-4. Its hard to tell be it seems that this is true in many of the super regions ...  ADMIXTURE tends use fewer factors to explain population structure in each region, leading to a more clustered result? 

### SNP Weights

Lets take a closer look at the drift factors to see if they are clustering in particular regions of the genome. As a first pass I take the top 5% of SNPs weighted on each drift event (to be clear I ignore the sign of each SNP). I would like to use the lfsr here but I will return to later. I then made a Manhatten plot for each chromosome and factor:

```{r flash-ld-factors-chrom-viz, fig.width=8.5, fig.height=8}
delta_thresh_df = delta_df %>%
                  gather(K, value, -chrom, -pos, -rsid) %>%
                  filter(K %in% paste0(2:11)) %>% 
                  group_by(K) %>%
                  summarise(t_95 = quantile(abs(value), probs = .95)[1],
                            t_99 = quantile(abs(value), probs = .99)[1])

# gather the data.frame for plotting
delta_gath_df = delta_df %>% 
                gather(K, value, -chrom, -pos, -rsid) %>%
                filter(K %in% paste0(2:11)) %>% 
                inner_join(delta_thresh_df, on="K") %>%
                filter(abs(value) > t_95) %>%
                filter(chrom %in% 1:22)

p = ggplot(delta_gath_df, aes(x=pos, y=abs(value), color=factor(K))) + geom_point(size=.5, alpha=.7) + 
    scale_color_brewer(palette = "Set3") + 
    facet_grid(factor(K, levels=2:11)~factor(chrom, levels = 1:22), scales="free") +
    geom_hline(data=delta_gath_df %>% distinct(chrom, K, t_99), aes(yintercept = t_99), linetype="dashed") +
    theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(),
          panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
    guides(color=FALSE) +
    xlab("Position") + 
    ylab("|Factor|")
p
```

We can see there are some regions on the chromosomes that are peaky as well as some regions that have no "outliers" at all. I then took the top 5 outliers in each factor and annotated them with some functional information:

```{r tophits}
grch37.snp = useMart(biomart="ENSEMBL_MART_SNP", host="grch37.ensembl.org", path="/biomart/martservice",dataset="hsapiens_snp")
grch37 = useMart(biomart="ENSEMBL_MART_ENSEMBL", host="grch37.ensembl.org", path="/biomart/martservice", dataset="hsapiens_gene_ensembl")

delta_tophit_df = delta_df %>%
                  gather(K, value, -chrom, -pos, -rsid) %>%
                  filter(K %in% paste0(2:21)) %>% 
                  group_by(K) %>%
                  top_n(5, abs(value))

table1 <- getBM(attributes=c('refsnp_id', 'chrom_start', 'minor_allele_freq', 'ensembl_gene_stable_id',
                             'consequence_type_tv', "associated_gene"), 
                filters = "snp_filter", 
                values = delta_tophit_df$rsid, 
                mart = grch37.snp)

table1$ensembl_gene_id = table1$ensembl_gene_stable_id
table2 <- getBM(attributes = c("ensembl_gene_id", "external_gene_name", "description"),
                filters = "ensembl_gene_id", 
                values =  table1$ensembl_gene_stable_id, 
                mart = grch37)

anno_df = table1 %>% left_join(table2, on="ensembl_gene_id") %>% mutate(rsid=refsnp_id) %>% 
          inner_join(delta_tophit_df, on="rsid") 

print(unique(anno_df$external_gene_name))
kable(anno_df %>% dplyr::select(rsid, chrom, pos, external_gene_name, K, value) %>% arrange(desc(K, value)))
```

We can see many of these top outliers are in genes (but we have to consider the array design to know if this is unusual). Its cool to see a couple very interesting genes pop up including HERC2 (eye color) and SLC45A2 (skin color) which have been previously studied for their selection signatures. Its also fun to look at some of the `rsids` on `https://popgen.uchicago.edu/ggv/` to get a sense of what kind of allele frequency distributions define each factor.
