---
title: "Human Origins Array Global Results"
author: "jhmarcus"
date: "2019-02-15"
output:
  workflowr::wflow_html:
    code_folding: show
---

## Imports

Lets import some needed packages:

```{r imports, warning=FALSE, message=FALSE}
library(ggplot2)
library(tidyr)
library(dplyr)
library(RColorBrewer)
source("../code/viz.R")
```

## Human Origins Global (LD Pruned)

This is the full Human Origins dataset 2068 sampled from around the world. I filtered out rare variants with global minor allele frequency less than 5%, and remove any variants with a missingness level greater than 1%. I then LD pruned the SNPs using standard parameters in `plink`, resulting in 167178 SNPs.

### Greedy

Lets first read the greedy `flashier` fit

```{r flash-greedy-ld}
flash_fit = readRDS("../output/flash_greedy/hoa_global_ld/HumanOriginsPublic2068_maf_geno_ldprune.rds")
K = ncol(flash_fit$loadings$normalized.loadings[[1]]) 
n = nrow(flash_fit$loadings$normalized.loadings[[1]])
p = nrow(flash_fit$loadings$normalized.loadings[[2]])
print(K)
print(n)
print(p)
```

Lets now plot the distribution of factors for each drift event

```{r flash-greedy-ld-viz-factors, warning=FALSE, message=FALSE, fig.width=8, fig.height=6}
# read factors
delta_df = as.data.frame(flash_fit$loadings$normalized.loadings[[2]])
colnames(delta_df)[1:K] = 1:K 

# gather the data.frame for plotting
delta_gath_df = delta_df %>% 
                gather(K, value) %>%
                filter(K!=1)

# plot the factors
K_ = K
p_fct = ggplot(delta_gath_df, aes(x=value)) + 
        scale_fill_manual(values = getPalette(K_)) +
        geom_histogram() + 
        facet_wrap(~factor(K, levels=2:K_), scales = "free") + 
        labs(fill="K") + 
        scale_x_continuous(breaks = scales::pretty_breaks(n = 3)) +
        scale_y_continuous(breaks = scales::pretty_breaks(n = 3)) + 
        theme_bw()
p_fct
```

We can see the later factors tend to get sparser but they still seem to contribute! Here is a plot of the "proportion of variance explained" of each factor:

```{r flash-greedy-ld-viz-pve, warning=FALSE, message=FALSE}
qplot(2:K, flash_fit$pve[2:K]) + ylab("Proportion of Varaince Explained") + xlab("K") + theme_bw()
print(flash_fit$pve)
```

It looks like the PVE drops off at around 11 or so? I setup the `flashier` run so it estimates a SNP specific precision term. Here is a histogram of fitted variances:

```{r flash-greedy-ld-viz-var, warning=FALSE, message=FALSE}
qplot(1/flash_fit$fit$est.tau) + xlab("Estimated Variance") + ylab("Count") + theme_bw()
```

Lets now look the the fitted means:

```{r flash-greedy-ld-viz-mean, warning=FALSE, message=FALSE}
qplot(delta_df$`1`) + xlab("Estimated Mean") + ylab("Count") + theme_bw()
```

The mean seems smaller than I would have expected but thats probably because its normalized to have unit norm?! These plots looks about reasonable as each of the SNP variances should roughly be interpreted as average heterozygosity $\approx 2p(1-p)$? The mean term should roughly be interpreted as the mean minor allele frequency at the SNP and thus we should see a quadratic relationship with the estimated variance:

```{r flash-greedy-ld-viz-mv, warning=FALSE, message=FALSE}
d1 = flash_fit$loadings$scale.constant[1]
p_mv = qplot(sqrt(d1) * delta_df$`1`, 1/flash_fit$fit$est.tau, alpha=.3) + 
       xlab("Rescaled Estimated Mean") + ylab("Estimated Variance") + 
       scale_alpha(guide = "none") + 
       stat_function(fun = function(x){return(2*x*(1-x))}, color="red") + 
       xlim(0, .5) + 
       theme_bw()
p_mv
```

I need to better understand the `flashier` output but it seems that when I scale the estimated mean by the square root of the output scale constant and plot this rescaled mean against the estimated variance, most of the SNPs have a mean-variance relationship expected under a simple Binomial model for the genotypes i.e. $y_{ij} \sim Binomial(2, p_{ij})$. Though, there are indeed many SNPs that have a a much larger variance than expected under this simple "Hardy-Weinberg" model. I wonder if there is anything "special" going on with those SNPs (this is something to follow up on). Lets now take a look at the loadings. First we setup a data.frame that we can work with:

```{r flash-greedy-ld-data-loadings, warning=FALSE, message=FALSE}
# read the meta data
meta_df = read.table("../data/meta/HumanOriginsPublic2068_maf_geno_ldprune.meta", sep=" ", header=T)

# setup loadings data.frame
l_df = as.data.frame(flash_fit$loadings$normalized.loadings[[1]])
K = ncol(l_df)
l_df$iid = as.vector(meta_df$iid) # individual ids
l_df$clst = meta_df$clst # population labels
pops = unique(l_df$clst) # all unique pop labels

# join with the meta data
l_df = l_df %>% inner_join(meta_df, on="clst")
l_df = l_df %>% arrange(region, clst) # sort by region then by population
l_df$iid = factor(l_df$iid, levels = l_df$iid) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

head(l_df)
```

Its hard to find a color scale that can sufficiently visualize all of the loadings in a single plot. Instead I just split the loadings up into two plots (K=2,...,11) and (K=12,...,21). Lets first visualize loadings 2 through 12:

```{r flash-greedy-ld-viz-loadings-2-11, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=12}
# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, -iid, -clst, -region, -country, -lat, -lon, -clst2) %>% 
            filter(K %in% paste0(2:11))

# Africa
africa_pops = get_pops(meta_df, "Africa")
p_africa = positive_structure_plot(l_gath_df %>% filter(region == "Africa"), 
                                   africa_pops, colset="Set3", label_size=5) +
           ggtitle("Africa") + theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, "America")
p_america = positive_structure_plot(l_gath_df %>% filter(region == "America"), 
                                    america_pops, colset="Set3", label_size=5) + 
            ggtitle("America") + theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, "CentralAsiaSiberia")
p_central_asia_siberia = positive_structure_plot(l_gath_df %>% filter(region == "CentralAsiaSiberia"), 
                                                 central_asia_siberia_pops, colset="Set3", label_size=5) + 
                         ggtitle("CentralAsiaSiberia") + theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, "EastAsia")
p_east_asia = positive_structure_plot(l_gath_df %>% filter(region == "EastAsia"), 
                                      east_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("EastAsia") + theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, "SouthAsia")
p_south_asia= positive_structure_plot(l_gath_df %>% filter(region == "SouthAsia"),
                                      south_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("SouthAsia") + theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, "WestEurasia")
p_west_eurasia = positive_structure_plot(l_gath_df %>% filter(region == "WestEurasia"), 
                                         west_eurasia_pops, colset="Set3", label_size=5) + 
                 ggtitle("WestEurasia") + theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, "Oceania")
p_oceania = positive_structure_plot(l_gath_df %>% filter(region == "Oceania"), 
                                    oceania_pops, colset="Set3", label_size=5) + 
            ggtitle("Oceania") + theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = "v") 
p
```

Lets now visualize loadings 12 to 21 (be careful: there is no connection to the colors in the last plot):

```{r flash-greedy-ld-viz-loadings-12-21, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=12}
# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, -iid, -clst, -region, -country, -lat, -lon, -clst2) %>% 
            filter(K %in% paste0(12:21))

# Africa
africa_pops = get_pops(meta_df, "Africa")
p_africa = positive_structure_plot(l_gath_df %>% filter(region == "Africa"), 
                                   africa_pops, colset="Set3", label_size=5) +
           ggtitle("Africa") + theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, "America")
p_america = positive_structure_plot(l_gath_df %>% filter(region == "America"), 
                                    america_pops, colset="Set3", label_size=5) + 
            ggtitle("America") + theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, "CentralAsiaSiberia")
p_central_asia_siberia = positive_structure_plot(l_gath_df %>% filter(region == "CentralAsiaSiberia"), 
                                                 central_asia_siberia_pops, colset="Set3", label_size=5) + 
                         ggtitle("CentralAsiaSiberia") + theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, "EastAsia")
p_east_asia = positive_structure_plot(l_gath_df %>% filter(region == "EastAsia"), 
                                      east_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("EastAsia") + theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, "SouthAsia")
p_south_asia= positive_structure_plot(l_gath_df %>% filter(region == "SouthAsia"),
                                      south_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("SouthAsia") + theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, "WestEurasia")
p_west_eurasia = positive_structure_plot(l_gath_df %>% filter(region == "WestEurasia"), 
                                         west_eurasia_pops, colset="Set3", label_size=5) + 
                 ggtitle("WestEurasia") + theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, "Oceania")
p_oceania = positive_structure_plot(l_gath_df %>% filter(region == "Oceania"), 
                                    oceania_pops, colset="Set3", label_size=5) + 
            ggtitle("Oceania") + theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = "v") 
p
```

Its kinda interesting to see that some populations have zero loading on later factors. This would be difficult to visualize see if using a single plot for all the factors.

### ADMIXTURE

Lets visualize ADMIXTURE with 9 factors which should roughly align to the first plot i.e. `FLASH` with 2,...,11 (be careful: there is no connection to the colors in the last plot):

```{r admixtrue-ld-viz-loadings-9, warning=FALSE, message=FALSE, fig.width=8.5, fig.height=12}
l_df = read.table("../output/admixture/hoa_global_ld/HumanOriginsPublic2068_maf_geno_ldprune.K9r1.Q", sep=" ", header=F)
K = ncol(l_df)

l_df$iid = as.vector(meta_df$iid) # individual ids
l_df$clst = meta_df$clst # population labels

# join with the meta data
l_df = l_df %>% inner_join(meta_df, on="clst")
l_df = l_df %>% arrange(region, clst) # sort by region then by population
l_df$iid = factor(l_df$iid, levels = l_df$iid) # make sure the ids are sorted
colnames(l_df)[1:K] = 1:K

# gather the data.frame for plotting
l_gath_df = l_df %>% 
            gather(K, value, -iid, -clst, -region, -country, -lat, -lon, -clst2)

# Africa
africa_pops = get_pops(meta_df, "Africa")
p_africa = positive_structure_plot(l_gath_df %>% filter(region == "Africa"), 
                                   africa_pops, colset="Set3", label_size=5) +
           ggtitle("Africa") + theme(plot.title = element_text(size=6))

# America
america_pops = get_pops(meta_df, "America")
p_america = positive_structure_plot(l_gath_df %>% filter(region == "America"), 
                                    america_pops, colset="Set3", label_size=5) + 
            ggtitle("America") + theme(plot.title = element_text(size=6))

# Central Asia Siberia
central_asia_siberia_pops = get_pops(meta_df, "CentralAsiaSiberia")
p_central_asia_siberia = positive_structure_plot(l_gath_df %>% filter(region == "CentralAsiaSiberia"), 
                                                 central_asia_siberia_pops, colset="Set3", label_size=5) + 
                         ggtitle("CentralAsiaSiberia") + theme(plot.title = element_text(size=6))

# East Asia
east_asia_pops = get_pops(meta_df, "EastAsia")
p_east_asia = positive_structure_plot(l_gath_df %>% filter(region == "EastAsia"), 
                                      east_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("EastAsia") + theme(plot.title = element_text(size=6))

# South Asia
south_asia_pops = get_pops(meta_df, "SouthAsia")
p_south_asia= positive_structure_plot(l_gath_df %>% filter(region == "SouthAsia"),
                                      south_asia_pops, colset="Set3", label_size=5) + 
              ggtitle("SouthAsia") + theme(plot.title = element_text(size=6))

# West Eurasia
west_eurasia_pops = get_pops(meta_df, "WestEurasia")
p_west_eurasia = positive_structure_plot(l_gath_df %>% filter(region == "WestEurasia"), 
                                         west_eurasia_pops, colset="Set3", label_size=5) + 
                 ggtitle("WestEurasia") + theme(plot.title = element_text(size=6))

# Oceania
oceania_pops = get_pops(meta_df, "Oceania")
p_oceania = positive_structure_plot(l_gath_df %>% filter(region == "Oceania"), 
                                    oceania_pops, colset="Set3", label_size=5) + 
            ggtitle("Oceania") + theme(plot.title = element_text(size=6))

# Global
p = cowplot::plot_grid(p_africa, p_west_eurasia, p_central_asia_siberia, p_america, p_east_asia, p_south_asia, p_oceania, 
                       rel_heights = c(1.2, 1.3, 1, 1, 1, 1, 1.1),
                       nrow = 7, align = "v") 
p
```

There is a lot that one can compare between the ADMIXTURE and FLASH results. A high level observation seems that the ADMIXTURE results look a bit more clustered i.e. the Americas and East Asia look like they are explained mostly by 1 or 2 factors whereas FLASH uses 3-4. Its hard to tell be it seems that this is true in many of the super regions ...  ADMIXTURE tends use fewer factors to explain population structure in each region, leading to a more clustered result? 